--- a/net/minecraft/server/level/ChunkTaskPriorityQueue.java
+++ b/net/minecraft/server/level/ChunkTaskPriorityQueue.java
@@ -17,7 +_,7 @@
 public class ChunkTaskPriorityQueue<T> {
     public static final int PRIORITY_LEVEL_COUNT = ChunkLevel.MAX_LEVEL + 2;
     private final List<Long2ObjectLinkedOpenHashMap<List<Optional<T>>>> taskQueue = IntStream.range(0, PRIORITY_LEVEL_COUNT)
-        .mapToObj(i -> new Long2ObjectLinkedOpenHashMap())
+        .mapToObj(i -> new Long2ObjectLinkedOpenHashMap<List<Optional<T>>>())
         .collect(Collectors.toList());
     private volatile int firstQueue = PRIORITY_LEVEL_COUNT;
     private final String name;
@@ -94,7 +_,7 @@
                 this.firstQueue++;
             }
 
-            return list.stream().map(optional -> optional.map(Either::left).orElseGet(() -> Either.right(this.acquire(l))));
+            return list.stream().map(optional -> optional.<Either<T,Runnable>>map(Either::left).orElseGet(() -> Either.right(this.acquire(l))));
         }
     }
 
