--- a/net/minecraft/world/ticks/BlackholeTickAccess.java
+++ b/net/minecraft/world/ticks/BlackholeTickAccess.java
@@ -40,10 +40,10 @@
     };
 
     public static <T> TickContainerAccess<T> emptyContainer() {
-        return CONTAINER_BLACKHOLE;
+        return (TickContainerAccess<T>) CONTAINER_BLACKHOLE;
     }
 
     public static <T> LevelTickAccess<T> emptyLevelList() {
-        return LEVEL_BLACKHOLE;
+        return (LevelTickAccess<T>) LEVEL_BLACKHOLE;
     }
 }
