--- a/net/minecraft/server/commands/EnchantCommand.java
+++ b/net/minecraft/server/commands/EnchantCommand.java
@@ -100,12 +100,12 @@
                 if (targets.size() == 1) {
                     source.sendSuccess(
                         () -> Component.translatable(
-                                "commands.enchant.success.single", enchantment.getFullname(level), targets.iterator().next().getDisplayName()
+                                "commands.enchant.success.single", enchantmentx.getFullname(level), targets.iterator().next().getDisplayName()
                             ),
                         true
                     );
                 } else {
-                    source.sendSuccess(() -> Component.translatable("commands.enchant.success.multiple", enchantment.getFullname(level), targets.size()), true);
+                    source.sendSuccess(() -> Component.translatable("commands.enchant.success.multiple", enchantmentx.getFullname(level), targets.size()), true);
                 }
 
                 return i;
