--- a/net/minecraft/core/registries/BuiltInRegistries.java
+++ b/net/minecraft/core/registries/BuiltInRegistries.java
@@ -284,7 +284,7 @@
     ) {
         ResourceLocation resourcelocation = key.location();
         LOADERS.put(resourcelocation, () -> initializer.run(registry));
-        WRITABLE_REGISTRY.register(key, registry, lifecycle);
+        WRITABLE_REGISTRY.register((ResourceKey) key, registry, lifecycle);
         return registry;
     }
 
