--- a/net/minecraft/server/dedicated/DedicatedServerProperties.java
+++ b/net/minecraft/server/dedicated/DedicatedServerProperties.java
@@ -185,7 +185,7 @@
     }
 
     private static FeatureFlagSet getFeatures(String string) {
-        return FeatureFlags.REGISTRY.fromNames(COMMA_SPLITTER.splitToStream(string).mapMulti((string1, consumer) -> {
+        return FeatureFlags.REGISTRY.fromNames(COMMA_SPLITTER.splitToStream(string).<ResourceLocation>mapMulti((string1, consumer) -> {
             ResourceLocation resourceLocation = ResourceLocation.tryParse(string1);
             if (resourceLocation == null) {
                 LOGGER.warn("Invalid resource location {}, ignoring", string1);
