--- a/net/minecraft/util/profiling/jfr/stats/TimedStatSummary.java
+++ b/net/minecraft/util/profiling/jfr/stats/TimedStatSummary.java
@@ -19,9 +19,9 @@
             T timedStat = list.get(0);
             T timedStat1 = list.get(list.size() - 1);
             T timedStat2 = list.size() > 1 ? list.get(list.size() - 2) : null;
-            int size = list.size();
+            int i = list.size();
             Map<Integer, Double> map = Percentiles.evaluate(list.stream().mapToLong(sample -> sample.duration().toNanos()).toArray());
-            return new TimedStatSummary<>(timedStat, timedStat1, timedStat2, size, map, duration);
+            return new TimedStatSummary<>(timedStat, timedStat1, timedStat2, i, map, duration);
         }
     }
 }
