--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -507,9 +507,10 @@
 
         int max = Math.max(0, this.unloadQueue.size() - 2000);
 
-        while((shouldKeepTicking.getAsBoolean() || max > 0) && (l = (long)this.unloadQueue.poll()) != null) {
+        Runnable runnable;
+        while((shouldKeepTicking.getAsBoolean() || max > 0) && (runnable = this.unloadQueue.poll()) != null) {
             --max;
-            l.run();
+            runnable.run();
         }
 
         int i1 = 0;
@@ -594,7 +595,7 @@
                 }
 
                 return isChunkDataValid;
-            })).thenApplyAsync(nbt -> {
+            })).<Either<ChunkAccess, ChunkHolder.ChunkLoadingFailure>>thenApplyAsync(nbt -> {
             this.level.getProfiler().incrementCounter("chunkLoad");
             if (nbt.isPresent()) {
                 ChunkAccess chunkAccess = ChunkSerializer.read(this.level, this.poiManager, pos, nbt.get());
