--- a/net/minecraft/util/datafix/fixes/ItemSpawnEggFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemSpawnEggFix.java
@@ -117,8 +117,9 @@
                         Typed<?> typed2 = typed.getOrCreateTyped(opticFinder2);
                         Typed<?> typed3 = typed2.getOrCreateTyped(opticFinder3);
                         Typed<?> typed4 = typed3.getOrCreateTyped(opticFinder4);
+                        Dynamic<?> f_Dynamic = dynamic;
                         Typed<?> typed5 = (Typed)typed4.write()
-                            .flatMap(dynamic2 -> typeRaw.readTyped(dynamic2.set("id", dynamic.createString(string))))
+                            .flatMap(dynamic2 -> typeRaw.readTyped(dynamic2.set("id", f_Dynamic.createString(string))))
                             .result()
                             .orElseThrow(() -> new IllegalStateException("Could not parse new entity"))
                             .getFirst();
