--- a/net/minecraft/util/datafix/fixes/ItemWrittenBookPagesStrictJsonFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemWrittenBookPagesStrictJsonFix.java
@@ -29,7 +29,7 @@
                             stream -> stream.map(
                                     dynamic1 -> {
                                         if (dynamic1.asString().result().isEmpty()) {
-                                            return (Dynamic)dynamic1;
+                                            return dynamic1;
                                         } else {
                                             String string = dynamic1.asString("");
                                             Component component = null;
