--- a/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplateManager.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplateManager.java
@@ -159,8 +159,8 @@
         if (!Files.isDirectory(structuresDirectoryPath)) {
             return Stream.empty();
         } else {
-            int len = extension.length();
-            Function<String, String> function = filename -> filename.substring(0, filename.length() - len);
+            int i = extension.length();
+            Function<String, String> function = filename -> filename.substring(0, filename.length() - i);
 
             try {
                 return Files.walk(structuresDirectoryPath).filter(path -> path.toString().endsWith(extension)).mapMulti((path, consumer) -> {
