--- a/net/minecraft/data/tags/TagsProvider.java
+++ b/net/minecraft/data/tags/TagsProvider.java
@@ -63,6 +63,8 @@
 
     @Override
     public CompletableFuture<?> run(CachedOutput output) {
+        record CombinedData<T>(HolderLookup.Provider contents, TagsProvider.TagLookup<T> parent) {
+        }
         return this.createContentsProvider()
             .thenApply(provider -> {
                 this.contentsDone.complete(null);
@@ -109,9 +111,6 @@
                 }
             );
 
-        record CombinedData<T>(HolderLookup.Provider contents, TagsProvider.TagLookup<T> parent) {
-        }
-
     }
 
     protected TagsProvider.TagAppender<T> tag(TagKey<T> tag) {
@@ -179,7 +178,7 @@
         }
 
         default boolean contains(TagKey<T> key) {
-            return this.apply((T)key).isPresent();
+            return this.apply(key).isPresent();
         }
     }
 }
