--- a/net/minecraft/world/entity/monster/piglin/StartHuntingHoglin.java
+++ b/net/minecraft/world/entity/monster/piglin/StartHuntingHoglin.java
@@ -10,7 +10,7 @@
 public class StartHuntingHoglin {
     public static OneShot<Piglin> create() {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.NEAREST_VISIBLE_HUNTABLE_HOGLIN),
                         instance.absent(MemoryModuleType.ANGRY_AT),
                         instance.absent(MemoryModuleType.HUNTED_RECENTLY),
@@ -20,14 +20,14 @@
                         instance,
                         (nearestVisibleHuntableHoglin, angryAt, huntedRecently, nearestVisibleAdultPiglins) -> (world, entity, time) -> {
                                 if (!entity.isBaby()
-                                    && !instance.<List>tryGet(nearestVisibleAdultPiglins)
+                                    && !instance.tryGet(nearestVisibleAdultPiglins)
                                         .map(list -> list.stream().anyMatch(StartHuntingHoglin::hasHuntedRecently))
                                         .isPresent()) {
                                     Hoglin hoglin = instance.get(nearestVisibleHuntableHoglin);
                                     PiglinAi.setAngerTarget(entity, hoglin);
                                     PiglinAi.dontKillAnyMoreHoglinsForAWhile(entity);
                                     PiglinAi.broadcastAngerTarget(entity, hoglin);
-                                    instance.<List>tryGet(nearestVisibleAdultPiglins)
+                                    instance.tryGet(nearestVisibleAdultPiglins)
                                         .ifPresent(list -> list.forEach(PiglinAi::dontKillAnyMoreHoglinsForAWhile));
                                     return true;
                                 } else {
