--- a/net/minecraft/world/entity/monster/piglin/StartAdmiringItemIfSeen.java
+++ b/net/minecraft/world/entity/monster/piglin/StartAdmiringItemIfSeen.java
@@ -10,7 +10,7 @@
 public class StartAdmiringItemIfSeen {
     public static BehaviorControl<LivingEntity> create(int i) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.present(MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM),
                         context.absent(MemoryModuleType.ADMIRING_ITEM),
                         context.absent(MemoryModuleType.ADMIRING_DISABLED),
