--- a/net/minecraft/world/entity/monster/piglin/StopAdmiringIfItemTooFarAway.java
+++ b/net/minecraft/world/entity/monster/piglin/StopAdmiringIfItemTooFarAway.java
@@ -11,7 +11,7 @@
 public class StopAdmiringIfItemTooFarAway<E extends Piglin> {
     public static BehaviorControl<LivingEntity> create(int i) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.present(MemoryModuleType.ADMIRING_ITEM), context.registered(MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM)
                     )
                     .apply(context, (admiringItem, nearestVisibleWantedItem) -> (world, entity, time) -> {
