--- a/net/minecraft/core/component/TypedDataComponent.java
+++ b/net/minecraft/core/component/TypedDataComponent.java
@@ -12,7 +_,7 @@
         @Override
         public TypedDataComponent<?> decode(RegistryFriendlyByteBuf registryFriendlyByteBuf) {
             DataComponentType<?> dataComponentType = DataComponentType.STREAM_CODEC.decode(registryFriendlyByteBuf);
-            return decodeTyped(registryFriendlyByteBuf, (DataComponentType<T>)dataComponentType);
+            return decodeTyped(registryFriendlyByteBuf, dataComponentType);
         }
 
         private static <T> TypedDataComponent<T> decodeTyped(RegistryFriendlyByteBuf registryFriendlyByteBuf, DataComponentType<T> dataComponentType) {
@@ -21,7 +_,7 @@
 
         @Override
         public void encode(RegistryFriendlyByteBuf registryFriendlyByteBuf, TypedDataComponent<?> typedDataComponent) {
-            encodeCap(registryFriendlyByteBuf, (TypedDataComponent<T>)typedDataComponent);
+            encodeCap(registryFriendlyByteBuf, typedDataComponent);
         }
 
         private static <T> void encodeCap(RegistryFriendlyByteBuf registryFriendlyByteBuf, TypedDataComponent<T> typedDataComponent) {
