--- a/net/minecraft/world/level/biome/Climate.java
+++ b/net/minecraft/world/level/biome/Climate.java
@@ -267,7 +267,7 @@
                     throw new IllegalStateException("Expecting parameter space to be 7, got " + i);
                 } else {
                     List<Climate.RTree.Leaf<T>> list = entries.stream()
-                        .map(entry -> new Climate.RTree.Leaf(entry.getFirst(), entry.getSecond()))
+                        .map(entry -> new Climate.RTree.Leaf<>(entry.getFirst(), entry.getSecond()))
                         .collect(Collectors.toCollection(ArrayList::new));
                     return new Climate.RTree<>(build(i, list));
                 }
