--- a/net/minecraft/data/info/RegistryDumpReport.java
+++ b/net/minecraft/data/info/RegistryDumpReport.java
@@ -34,7 +34,7 @@
             jsonObject.addProperty("default", defaultKey.toString());
         }
 
-        int defaultKey1 = BuiltInRegistries.REGISTRY.getId(registry1);
+        int defaultKey1 = ((Registry) BuiltInRegistries.REGISTRY).getId(registry1);
         jsonObject.addProperty("protocol_id", defaultKey1);
         JsonObject jsonObject1 = new JsonObject();
         registry1.holders().forEach(entry -> {
