--- a/net/minecraft/data/info/RegistryDumpReport.java
+++ b/net/minecraft/data/info/RegistryDumpReport.java
@@ -34,7 +34,7 @@
             jsonObject.addProperty("default", defaultKey.toString());
         }
 
-        int id = BuiltInRegistries.REGISTRY.getId(registry);
+        int id = ((Registry) BuiltInRegistries.REGISTRY).getId(registry);
         jsonObject.addProperty("protocol_id", id);
         JsonObject jsonObject1 = new JsonObject();
         registry.holders().forEach(holder -> {
