--- a/net/minecraft/world/inventory/RecipeBookMenu.java
+++ b/net/minecraft/world/inventory/RecipeBookMenu.java
@@ -13,7 +13,7 @@
     }
 
     public void handlePlacement(boolean flag, RecipeHolder<?> recipeHolder, ServerPlayer serverPlayer) {
-        new ServerPlaceRecipe<>(this).recipeClicked(serverPlayer, recipeHolder, flag);
+        new ServerPlaceRecipe<>(this).recipeClicked(serverPlayer, (RecipeHolder<? extends Recipe<C>>) recipeHolder, flag);
     }
 
     public abstract void fillCraftSlotsStackedContents(StackedContents itemHelper);
