--- a/net/minecraft/util/datafix/fixes/ChunkHeightAndBiomeFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkHeightAndBiomeFix.java
@@ -222,7 +222,8 @@
         } else if (ints != null && ints.length == 1024) {
             for (int i1 = 0; i1 < 16; i1++) {
                 int i2 = i1 - i;
-                dynamics[i2] = makeBiomeContainer(dynamic, i4 -> getOldBiome(ints, i1 * 64 + i4));
+                final int f_i = i1;
+                dynamics[i2] = makeBiomeContainer(dynamic, i4 -> getOldBiome(ints, f_i * 64 + i4));
             }
 
             if (overworld) {
