--- a/net/minecraft/network/protocol/configuration/ClientboundUpdateEnabledFeaturesPacket.java
+++ b/net/minecraft/network/protocol/configuration/ClientboundUpdateEnabledFeaturesPacket.java
@@ -7,8 +7,9 @@
 import net.minecraft.resources.ResourceLocation;
 
 public record ClientboundUpdateEnabledFeaturesPacket(Set<ResourceLocation> features) implements Packet<ClientConfigurationPacketListener> {
-    public ClientboundUpdateEnabledFeaturesPacket(FriendlyByteBuf buffer) {
-        this(buffer.readCollection(HashSet::new, FriendlyByteBuf::readResourceLocation));
+
+    public ClientboundUpdateEnabledFeaturesPacket(FriendlyByteBuf friendlyByteBuf) {
+        this(friendlyByteBuf.<ResourceLocation, Set<ResourceLocation>>readCollection(HashSet::new, FriendlyByteBuf::readResourceLocation));
     }
 
     @Override
