--- a/net/minecraft/core/RegistryAccess.java
+++ b/net/minecraft/core/RegistryAccess.java
@@ -100,7 +100,7 @@
         }
 
         private static <T> RegistryAccess.RegistryEntry<T> fromUntyped(ResourceKey<? extends Registry<?>> key, Registry<?> value) {
-            return new RegistryAccess.RegistryEntry<>(key, (Registry<T>)value);
+            return new RegistryAccess.RegistryEntry<>((ResourceKey<Registry<T>>) key, (Registry<T>)value);
         }
 
         private RegistryAccess.RegistryEntry<T> freeze() {
