--- a/net/minecraft/world/entity/ai/behavior/BabyFollowAdult.java
+++ b/net/minecraft/world/entity/ai/behavior/BabyFollowAdult.java
@@ -17,7 +17,7 @@
 
     public static OneShot<AgeableMob> create(UniformInt followRange, Function<LivingEntity, Float> speedModifier) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.NEAREST_VISIBLE_ADULT),
                         instance.registered(MemoryModuleType.LOOK_TARGET),
                         instance.absent(MemoryModuleType.WALK_TARGET)
