--- a/net/minecraft/world/entity/ai/behavior/LocateHidingPlace.java
+++ b/net/minecraft/world/entity/ai/behavior/LocateHidingPlace.java
@@ -15,7 +15,7 @@
 public class LocateHidingPlace {
     public static OneShot<LivingEntity> create(int radius, float speedModifier, int closeEnoughDist) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.absent(MemoryModuleType.WALK_TARGET),
                         instance.registered(MemoryModuleType.HOME),
                         instance.registered(MemoryModuleType.HIDING_PLACE),
