--- a/net/minecraft/network/chat/contents/ScoreContents.java
+++ b/net/minecraft/network/chat/contents/ScoreContents.java
@@ -107,8 +107,8 @@
 
     @Override
     public int hashCode() {
-        int hashCode = this.name.hashCode();
-        return 31 * hashCode + this.objective.hashCode();
+        int i = this.name.hashCode();
+        return 31 * i + this.objective.hashCode();
     }
 
     @Override
