--- a/net/minecraft/network/chat/contents/TranslatableContents.java
+++ b/net/minecraft/network/chat/contents/TranslatableContents.java
@@ -179,9 +179,9 @@
 
     @Override
     public int hashCode() {
-        int hashCode = Objects.hashCode(this.key);
-        hashCode = 31 * hashCode + Objects.hashCode(this.fallback);
-        return 31 * hashCode + Arrays.hashCode(this.args);
+        int i = Objects.hashCode(this.key);
+        i = 31 * i + Objects.hashCode(this.fallback);
+        return 31 * i + Arrays.hashCode(this.args);
     }
 
     @Override
