--- a/net/minecraft/nbt/NbtUtils.java
+++ b/net/minecraft/nbt/NbtUtils.java
@@ -375,7 +375,7 @@
                     indent(indentLevel + 1, stringBuilder);
                 }
 
-                int ix = list3.stream().mapToInt(String::length).max().orElse(0);
+                ix = list3.stream().mapToInt(String::length).max().orElse(0);
                 String stringx = Strings.repeat(" ", ix);
 
                 for (int i2 = 0; i2 < list3.size(); i2++) {
@@ -402,7 +402,7 @@
             case 11:
                 IntArrayTag list2 = (IntArrayTag)tag;
                 int[] asIntArray = list2.getAsIntArray();
-                int ix = 0;
+                ix = 0;
 
                 for (int i3 : asIntArray) {
                     ix = Math.max(ix, String.format(Locale.ROOT, "%X", i3).length());
