--- a/net/minecraft/world/attribute/modifier/ColorModifier.java
+++ b/net/minecraft/world/attribute/modifier/ColorModifier.java
@@ -26,10 +_,10 @@
             return LerpFunction.ofColor();
         }
     };
-    ColorModifier<Integer> ADD = ARGB::addRgb;
-    ColorModifier<Integer> SUBTRACT = ARGB::subtractRgb;
-    ColorModifier<Integer> MULTIPLY_RGB = ARGB::multiply;
-    ColorModifier<Integer> MULTIPLY_ARGB = ARGB::multiply;
+    ColorModifier<Integer> ADD = (ColorModifier.RgbModifier)ARGB::addRgb;
+    ColorModifier<Integer> SUBTRACT = (ColorModifier.RgbModifier)ARGB::subtractRgb;
+    ColorModifier<Integer> MULTIPLY_RGB = (ColorModifier.RgbModifier)ARGB::multiply;
+    ColorModifier<Integer> MULTIPLY_ARGB = (ColorModifier.ArgbModifier)ARGB::multiply;
     ColorModifier<ColorModifier.BlendToGray> BLEND_TO_GRAY = new ColorModifier<ColorModifier.BlendToGray>() {
         @Override
         public Integer apply(final Integer subject, final ColorModifier.BlendToGray argument) {
