--- a/net/minecraft/network/syncher/EntityDataSerializer.java
+++ b/net/minecraft/network/syncher/EntityDataSerializer.java
@@ -19,12 +19,12 @@
         return new EntityDataSerializer.ForValueType<T>() {
             @Override
             public void write(FriendlyByteBuf buf, T object) {
-                writer1.accept((T)buf, object);
+                writer1.accept(buf, object);
             }
 
             @Override
             public T read(FriendlyByteBuf friendlyByteBuf) {
-                return reader.apply((T)friendlyByteBuf);
+                return reader.apply(friendlyByteBuf);
             }
         };
     }
