--- a/net/minecraft/world/entity/monster/warden/WardenAi.java
+++ b/net/minecraft/world/entity/monster/warden/WardenAi.java
@@ -83,7 +83,7 @@
         MemoryModuleType.SONIC_BOOM_SOUND_DELAY
     );
     private static final BehaviorControl<Warden> DIG_COOLDOWN_SETTER = BehaviorBuilder.create(
-        instance -> instance.<MemoryAccessor>group(instance.registered(MemoryModuleType.DIG_COOLDOWN))
+        instance -> instance.group(instance.registered(MemoryModuleType.DIG_COOLDOWN))
                 .apply(instance, memoryAccessor -> (serverLevel, warden, l) -> {
                         if (instance.tryGet(memoryAccessor).isPresent()) {
                             memoryAccessor.setWithExpiry(Unit.INSTANCE, 1200L);
