--- a/net/minecraft/world/level/levelgen/Beardifier.java
+++ b/net/minecraft/world/level/levelgen/Beardifier.java
@@ -138,8 +138,8 @@
     }
 
     private static double getBuryContribution(int x, int y, int z) {
-        double len = Mth.length((double)x, (double)y / 2.0, (double)z);
-        return Mth.clampedMap(len, 0.0, 6.0, 1.0, 0.0);
+        double d = Mth.length((double)x, (double)y / 2.0, (double)z);
+        return Mth.clampedMap(d, 0.0, 6.0, 1.0, 0.0);
     }
 
     private static double getBeardContribution(int x, int y, int z, int yy) {
