--- a/net/minecraft/world/level/levelgen/LegacyRandomSource.java
+++ b/net/minecraft/world/level/levelgen/LegacyRandomSource.java
@@ -69,8 +69,8 @@
 
         @Override
         public RandomSource fromHashOf(String seed) {
-            int hashCode = seed.hashCode();
-            return new LegacyRandomSource((long)hashCode ^ this.seed);
+            int i = seed.hashCode();
+            return new LegacyRandomSource((long)i ^ this.seed);
         }
 
         @VisibleForTesting
