--- a/net/minecraft/world/level/levelgen/PatrolSpawner.java
+++ b/net/minecraft/world/level/levelgen/PatrolSpawner.java
@@ -37,20 +37,20 @@
                 } else if (randomSource.nextInt(5) != 0) {
                     return 0;
                 } else {
-                    int size = world.players().size();
-                    if (size < 1) {
+                    int i = world.players().size();
+                    if (i < 1) {
                         return 0;
                     } else {
-                        Player player = world.players().get(randomSource.nextInt(size));
+                        Player player = world.players().get(randomSource.nextInt(i));
                         if (player.isSpectator()) {
                             return 0;
                         } else if (world.isCloseToVillage(player.blockPosition(), 2)) {
                             return 0;
                         } else {
-                            int i = (24 + randomSource.nextInt(24)) * (randomSource.nextBoolean() ? -1 : 1);
                             int i1 = (24 + randomSource.nextInt(24)) * (randomSource.nextBoolean() ? -1 : 1);
-                            BlockPos.MutableBlockPos mutableBlockPos = player.blockPosition().mutable().move(i, 0, i1);
-                            int i2 = 10;
+                            int i2 = (24 + randomSource.nextInt(24)) * (randomSource.nextBoolean() ? -1 : 1);
+                            BlockPos.MutableBlockPos mutableBlockPos = player.blockPosition().mutable().move(i1, 0, i2);
+                            int i3 = 10;
                             if (!world.hasChunksAt(
                                 mutableBlockPos.getX() - 10, mutableBlockPos.getZ() - 10, mutableBlockPos.getX() + 10, mutableBlockPos.getZ() + 10
                             )) {
@@ -60,13 +60,13 @@
                                 if (biome.is(BiomeTags.WITHOUT_PATROL_SPAWNS)) {
                                     return 0;
                                 } else {
-                                    int i3 = 0;
-                                    int i4 = (int)Math.ceil((double)world.getCurrentDifficultyAt(mutableBlockPos).getEffectiveDifficulty()) + 1;
+                                    int i4 = 0;
+                                    int i5 = (int)Math.ceil((double)world.getCurrentDifficultyAt(mutableBlockPos).getEffectiveDifficulty()) + 1;
 
-                                    for(int i5 = 0; i5 < i4; ++i5) {
-                                        ++i3;
+                                    for(int i6 = 0; i6 < i5; ++i6) {
+                                        ++i4;
                                         mutableBlockPos.setY(world.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, mutableBlockPos).getY());
-                                        if (i5 == 0) {
+                                        if (i6 == 0) {
                                             if (!this.spawnPatrolMember(world, mutableBlockPos, randomSource, true)) {
                                                 break;
                                             }
@@ -78,7 +78,7 @@
                                         mutableBlockPos.setZ(mutableBlockPos.getZ() + randomSource.nextInt(5) - randomSource.nextInt(5));
                                     }
 
-                                    return i3;
+                                    return i4;
                                 }
                             }
                         }
