--- a/net/minecraft/world/level/levelgen/WorldDimensions.java
+++ b/net/minecraft/world/level/levelgen/WorldDimensions.java
@@ -187,9 +187,9 @@
         Lifecycle lifecycle = list.size() == VANILLA_DIMENSION_COUNT ? Lifecycle.stable() : Lifecycle.experimental();
         WritableRegistry<LevelStem> writableRegistry = new MappedRegistry<>(Registries.LEVEL_STEM, lifecycle);
         list.forEach(entry -> writableRegistry.register(entry.key, entry.value, entry.lifecycle()));
-        Registry<LevelStem> frozen = writableRegistry.freeze();
-        PrimaryLevelData.SpecialWorldProperty specialWorldProperty = specialWorldProperty(frozen);
-        return new WorldDimensions.Complete(frozen.freeze(), specialWorldProperty);
+        Registry<LevelStem> registry = writableRegistry.freeze();
+        PrimaryLevelData.SpecialWorldProperty specialWorldProperty = specialWorldProperty(registry);
+        return new WorldDimensions.Complete(registry.freeze(), specialWorldProperty);
     }
 
     public static record Complete(Registry<LevelStem> dimensions, PrimaryLevelData.SpecialWorldProperty specialWorldProperty) {
