--- a/net/minecraft/data/loot/BlockLootSubProvider.java
+++ b/net/minecraft/data/loot/BlockLootSubProvider.java
@@ -135,7 +135,7 @@
 
     protected LootTable.Builder createSingleItemTableWithSilkTouch(Block block, ItemLike item) {
         return createSilkTouchDispatchTable(
-            block, this.applyExplosionCondition(block, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(item))
+            block, this.applyExplosionCondition(block, LootItem.lootTableItem(item))
         );
     }
 
@@ -147,7 +147,7 @@
                     .add(
                         this.applyExplosionDecay(
                             item,
-                            (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(item)
+                            LootItem.lootTableItem(item)
                                 .apply(SetItemCountFunction.setCount(count))
                         )
                     )
@@ -158,7 +158,7 @@
         return createSilkTouchDispatchTable(
             block,
             this.applyExplosionDecay(
-                block, (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(item).apply(SetItemCountFunction.setCount(count))
+                block, LootItem.lootTableItem(item).apply(SetItemCountFunction.setCount(count))
             )
         );
     }
@@ -185,7 +185,7 @@
                     .add(
                         this.applyExplosionDecay(
                             block,
-                            (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(block)
+                            LootItem.lootTableItem(block)
                                 .apply(
                                     SetItemCountFunction.setCount(ConstantValue.exactly(2.0F))
                                         .when(
@@ -257,7 +257,7 @@
             block,
             this.applyExplosionDecay(
                 block,
-                (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.RAW_COPPER)
+                LootItem.lootTableItem(Items.RAW_COPPER)
                     .apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 5.0F)))
                     .apply(ApplyBonusCount.addOreBonusCount(Enchantments.BLOCK_FORTUNE))
             )
@@ -269,7 +269,7 @@
             block,
             this.applyExplosionDecay(
                 block,
-                (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.LAPIS_LAZULI)
+                LootItem.lootTableItem(Items.LAPIS_LAZULI)
                     .apply(SetItemCountFunction.setCount(UniformGenerator.between(4.0F, 9.0F)))
                     .apply(ApplyBonusCount.addOreBonusCount(Enchantments.BLOCK_FORTUNE))
             )
@@ -281,7 +281,7 @@
             block,
             this.applyExplosionDecay(
                 block,
-                (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.REDSTONE)
+                LootItem.lootTableItem(Items.REDSTONE)
                     .apply(SetItemCountFunction.setCount(UniformGenerator.between(4.0F, 5.0F)))
                     .apply(ApplyBonusCount.addUniformBonusCount(Enchantments.BLOCK_FORTUNE))
             )
@@ -350,7 +350,7 @@
             block,
             this.applyExplosionDecay(
                 block,
-                (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(item)
+                LootItem.lootTableItem(item)
                     .apply(ApplyBonusCount.addOreBonusCount(Enchantments.BLOCK_FORTUNE))
             )
         );
@@ -361,7 +361,7 @@
             block,
             this.applyExplosionDecay(
                 block,
-                (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(item)
+                LootItem.lootTableItem(item)
                     .apply(SetItemCountFunction.setCount(UniformGenerator.between(-6.0F, 2.0F)))
                     .apply(LimitCount.limitCount(IntRange.lowerBound(0)))
             )
@@ -373,7 +373,7 @@
             block,
             this.applyExplosionDecay(
                 block,
-                (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.WHEAT_SEEDS)
+                LootItem.lootTableItem(Items.WHEAT_SEEDS)
                     .when(LootItemRandomChanceCondition.randomChance(0.125F))
                     .apply(ApplyBonusCount.addUniformBonusCount(Enchantments.BLOCK_FORTUNE, 2))
             )
@@ -425,7 +425,7 @@
                     .add(
                         this.applyExplosionDecay(
                             block,
-                            (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(block)
+                            LootItem.lootTableItem(block)
                                 .when(builder)
                                 .apply(
                                     Direction.values(),
@@ -447,7 +447,7 @@
     protected LootTable.Builder createLeavesDrops(Block leavesBlock, Block saplingBlock, float... chances) {
         return createSilkTouchOrShearsDispatchTable(
                 leavesBlock,
-                this.applyExplosionCondition(leavesBlock, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(saplingBlock))
+                this.applyExplosionCondition(leavesBlock, LootItem.lootTableItem(saplingBlock))
                     .when(BonusLevelTableCondition.bonusLevelFlatChance(Enchantments.BLOCK_FORTUNE, chances))
             )
             .withPool(
@@ -457,7 +457,7 @@
                     .add(
                         this.applyExplosionDecay(
                                 leavesBlock,
-                                (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.STICK)
+                                LootItem.lootTableItem(Items.STICK)
                                     .apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 2.0F)))
                             )
                             .when(BonusLevelTableCondition.bonusLevelFlatChance(Enchantments.BLOCK_FORTUNE, NORMAL_LEAVES_STICK_CHANCES))
@@ -473,7 +473,7 @@
                     .when(HAS_NO_SHEARS_OR_SILK_TOUCH)
                     .add(
                         this.applyExplosionCondition(
-                                oakLeavesBlock, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.APPLE)
+                                oakLeavesBlock, LootItem.lootTableItem(Items.APPLE)
                             )
                             .when(
                                 BonusLevelTableCondition.bonusLevelFlatChance(Enchantments.BLOCK_FORTUNE, 0.005F, 0.0055555557F, 0.00625F, 0.008333334F, 0.025F)
@@ -487,7 +487,7 @@
             block,
             this.applyExplosionDecay(
                     Blocks.MANGROVE_LEAVES,
-                    (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.STICK)
+                    LootItem.lootTableItem(Items.STICK)
                         .apply(SetItemCountFunction.setCount(UniformGenerator.between(1.0F, 2.0F)))
                 )
                 .when(BonusLevelTableCondition.bonusLevelFlatChance(Enchantments.BLOCK_FORTUNE, NORMAL_LEAVES_STICK_CHANCES))
@@ -524,7 +524,7 @@
             .apply(SetItemCountFunction.setCount(ConstantValue.exactly(2.0F)))
             .when(HAS_SHEARS)
             .otherwise(
-                this.applyExplosionCondition(block, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.WHEAT_SEEDS))
+                this.applyExplosionCondition(block, LootItem.lootTableItem(Items.WHEAT_SEEDS))
                     .when(LootItemRandomChanceCondition.randomChance(0.125F))
             );
         return LootTable.lootTable()
@@ -576,7 +576,7 @@
                     .add(
                         this.applyExplosionDecay(
                             candleBlock,
-                            (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(candleBlock)
+                            LootItem.lootTableItem(candleBlock)
                                 .apply(
                                     List.of(2, 3, 4),
                                     integer -> SetItemCountFunction.setCount(ConstantValue.exactly((float)integer.intValue()))
@@ -600,7 +600,7 @@
                     .add(
                         this.applyExplosionDecay(
                             petalBlock,
-                            (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(petalBlock)
+                            LootItem.lootTableItem(petalBlock)
                                 .apply(
                                     IntStream.rangeClosed(1, 4).boxed().toList(),
                                     integer -> SetItemCountFunction.setCount(ConstantValue.exactly((float)integer.intValue()))
