--- a/net/minecraft/network/protocol/game/ClientboundChunkBatchStartPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundChunkBatchStartPacket.java
@@ -4,7 +4,7 @@
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.network.protocol.Packet;
 
-public final class ClientboundChunkBatchStartPacket extends Record implements Packet<ClientGamePacketListener> {
+public final class ClientboundChunkBatchStartPacket implements Packet<ClientGamePacketListener> {
     public ClientboundChunkBatchStartPacket(FriendlyByteBuf friendlyByteBuf) {
         this();
     }
@@ -20,19 +20,4 @@
     public void handle(ClientGamePacketListener handler) {
         handler.handleChunkBatchStart(this);
     }
-
-    @Override
-    public final String toString() {
-        return ObjectMethods.bootstrap<"toString",ClientboundChunkBatchStartPacket,"">(this);
-    }
-
-    @Override
-    public final int hashCode() {
-        return ObjectMethods.bootstrap<"hashCode",ClientboundChunkBatchStartPacket,"">(this);
-    }
-
-    @Override
-    public final boolean equals(Object object) {
-        return ObjectMethods.bootstrap<"equals",ClientboundChunkBatchStartPacket,"">(this, object);
-    }
 }
