--- a/net/minecraft/network/protocol/game/ClientboundStartConfigurationPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundStartConfigurationPacket.java
@@ -1,11 +1,10 @@
 package net.minecraft.network.protocol.game;
 
-import java.lang.runtime.ObjectMethods;
 import net.minecraft.network.ConnectionProtocol;
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.network.protocol.Packet;
 
-public final class ClientboundStartConfigurationPacket extends Record implements Packet<ClientGamePacketListener> {
+public record ClientboundStartConfigurationPacket() implements Packet<ClientGamePacketListener> {
     public ClientboundStartConfigurationPacket(FriendlyByteBuf friendlyByteBuf) {
         this();
     }
@@ -26,19 +25,4 @@
     public ConnectionProtocol nextProtocol() {
         return ConnectionProtocol.CONFIGURATION;
     }
-
-    @Override
-    public final String toString() {
-        return ObjectMethods.bootstrap<"toString",ClientboundStartConfigurationPacket,"">(this);
-    }
-
-    @Override
-    public final int hashCode() {
-        return ObjectMethods.bootstrap<"hashCode",ClientboundStartConfigurationPacket,"">(this);
-    }
-
-    @Override
-    public final boolean equals(Object object) {
-        return ObjectMethods.bootstrap<"equals",ClientboundStartConfigurationPacket,"">(this, object);
-    }
 }
