--- a/net/minecraft/tags/TagLoader.java
+++ b/net/minecraft/tags/TagLoader.java
@@ -85,7 +85,7 @@
             @Nullable
             @Override
             public T element(ResourceLocation elementLocation) {
-                return TagLoader.this.idToValue.apply(elementLocation).orElse((T)null);
+                return TagLoader.this.idToValue.apply(elementLocation).orElse(null);
             }
 
             @Nullable
