--- a/net/minecraft/world/entity/monster/piglin/RememberIfHoglinWasKilled.java
+++ b/net/minecraft/world/entity/monster/piglin/RememberIfHoglinWasKilled.java
@@ -10,7 +10,7 @@
 public class RememberIfHoglinWasKilled {
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.ATTACK_TARGET), instance.registered(MemoryModuleType.HUNTED_RECENTLY)
                     )
                     .apply(instance, (attackTarget, huntedRecently) -> (level, entity, gameTime) -> {
