--- a/net/minecraft/world/entity/monster/piglin/StartAdmiringItemIfSeen.java
+++ b/net/minecraft/world/entity/monster/piglin/StartAdmiringItemIfSeen.java
@@ -10,7 +10,7 @@
 public class StartAdmiringItemIfSeen {
     public static BehaviorControl<LivingEntity> create(int admireDuration) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM),
                         instance.absent(MemoryModuleType.ADMIRING_ITEM),
                         instance.absent(MemoryModuleType.ADMIRING_DISABLED),
