--- a/net/minecraft/world/entity/monster/piglin/StopAdmiringIfTiredOfTryingToReachItem.java
+++ b/net/minecraft/world/entity/monster/piglin/StopAdmiringIfTiredOfTryingToReachItem.java
@@ -10,7 +10,7 @@
 public class StopAdmiringIfTiredOfTryingToReachItem {
     public static BehaviorControl<LivingEntity> create(int maxTimeToReachItem, int disableDuration) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.ADMIRING_ITEM),
                         instance.present(MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM),
                         instance.registered(MemoryModuleType.TIME_TRYING_TO_REACH_ADMIRE_ITEM),
