--- a/net/minecraft/world/entity/monster/piglin/StopAdmiringIfItemTooFarAway.java
+++ b/net/minecraft/world/entity/monster/piglin/StopAdmiringIfItemTooFarAway.java
@@ -11,7 +11,7 @@
 public class StopAdmiringIfItemTooFarAway<E extends Piglin> {
     public static BehaviorControl<LivingEntity> create(int maxDist) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.ADMIRING_ITEM), instance.registered(MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM)
                     )
                     .apply(instance, (admiringItem, nearestVisibleWantedItem) -> (level, entity, gameTime) -> {
