--- a/net/minecraft/world/entity/monster/piglin/StopHoldingItemIfNoLongerAdmiring.java
+++ b/net/minecraft/world/entity/monster/piglin/StopHoldingItemIfNoLongerAdmiring.java
@@ -9,7 +9,7 @@
 public class StopHoldingItemIfNoLongerAdmiring {
     public static BehaviorControl<Piglin> create() {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor>group(context.absent(MemoryModuleType.ADMIRING_ITEM)).apply(context, admiringItem -> (world, entity, time) -> {
+            context -> context.group(context.absent(MemoryModuleType.ADMIRING_ITEM)).apply(context, admiringItem -> (world, entity, time) -> {
                         if (!entity.getOffhandItem().isEmpty() && !entity.getOffhandItem().is(Items.SHIELD)) {
                             PiglinAi.stopHoldingOffHandItem(entity, true);
                             return true;
