--- a/net/minecraft/world/entity/ai/behavior/MoveToSkySeeingSpot.java
+++ b/net/minecraft/world/entity/ai/behavior/MoveToSkySeeingSpot.java
@@ -16,7 +16,7 @@
 public class MoveToSkySeeingSpot {
     public static OneShot<LivingEntity> create(float speedModifier) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor>group(instance.absent(MemoryModuleType.WALK_TARGET))
+            instance -> instance.group(instance.absent(MemoryModuleType.WALK_TARGET))
                     .apply(instance, walkTarget -> (level, entity, gameTime) -> {
                             if (level.canSeeSky(entity.blockPosition())) {
                                 return false;
