--- a/net/minecraft/world/entity/ai/behavior/SetHiddenState.java
+++ b/net/minecraft/world/entity/ai/behavior/SetHiddenState.java
@@ -15,7 +15,7 @@
         int i = stayHiddenSeconds * 20;
         MutableInt mutableInt = new MutableInt(0);
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.HIDING_PLACE), instance.present(MemoryModuleType.HEARD_BELL_TIME)
                     )
                     .apply(instance, (hidingPlace, heardBellTime) -> (level, entity, gameTime) -> {
