--- a/net/minecraft/util/valueproviders/ConstantFloat.java
+++ b/net/minecraft/util/valueproviders/ConstantFloat.java
@@ -9,7 +9,7 @@
     public static final ConstantFloat ZERO = new ConstantFloat(0.0F);
     public static final Codec<ConstantFloat> CODEC = Codec.either(
             Codec.FLOAT,
-            RecordCodecBuilder.create(
+            RecordCodecBuilder.<ConstantFloat>create(
                 instance1 -> instance1.group(Codec.FLOAT.fieldOf("value").forGetter(constantFloat -> constantFloat.value)).apply(instance1, ConstantFloat::new)
             )
         )
