--- a/net/minecraft/util/valueproviders/ConstantInt.java
+++ b/net/minecraft/util/valueproviders/ConstantInt.java
@@ -9,7 +9,7 @@
     public static final ConstantInt ZERO = new ConstantInt(0);
     public static final Codec<ConstantInt> CODEC = Codec.either(
             Codec.INT,
-            RecordCodecBuilder.create(
+            RecordCodecBuilder.<ConstantInt>create(
                 instance1 -> instance1.group(Codec.INT.fieldOf("value").forGetter(constantInt -> constantInt.value)).apply(instance1, ConstantInt::new)
             )
         )
