--- a/net/minecraft/network/protocol/BundlerInfo.java
+++ b/net/minecraft/network/protocol/BundlerInfo.java
@@ -54,7 +54,7 @@
                         } else if (this.bundlePackets.size() >= 4096) {
                             throw new IllegalStateException("Too many packets in a bundle");
                         } else {
-                            this.bundlePackets.add(packet1);
+                            ((List) this.bundlePackets).add(packet1);
                             return null;
                         }
                     }
