--- a/net/minecraft/commands/arguments/EntityArgument.java
+++ b/net/minecraft/commands/arguments/EntityArgument.java
@@ -117,11 +117,10 @@
 
     @Override
     public <S> CompletableFuture<Suggestions> listSuggestions(CommandContext<S> context, SuggestionsBuilder suggestionsBuilder) {
-        StringReader source = context.getSource();
-        if (source instanceof SharedSuggestionProvider sharedSuggestionProvider) {
-            source = new StringReader(suggestionsBuilder.getInput());
-            source.setCursor(suggestionsBuilder.getStart());
-            EntitySelectorParser entitySelectorParser = new EntitySelectorParser(source, sharedSuggestionProvider.hasPermission(2));
+        if (context.getSource() instanceof SharedSuggestionProvider sharedSuggestionProvider) {
+            StringReader stringReader = new StringReader(suggestionsBuilder.getInput());
+            stringReader.setCursor(suggestionsBuilder.getStart());
+            EntitySelectorParser entitySelectorParser = new EntitySelectorParser(stringReader, sharedSuggestionProvider.hasPermission(2));
 
             try {
                 entitySelectorParser.parse();
