--- a/net/minecraft/commands/arguments/OperationArgument.java
+++ b/net/minecraft/commands/arguments/OperationArgument.java
@@ -57,7 +57,7 @@
     }
 
     private static OperationArgument.Operation getOperation(String string) throws CommandSyntaxException {
-        return (OperationArgument.Operation)(string.equals("><") ? (score, b) -> {
+        return (string.equals("><") ? (score, b) -> {
             int score1 = score.getScore();
             score.setScore(b.getScore());
             b.setScore(score1);
