--- a/net/minecraft/commands/arguments/ResourceOrTagArgument.java
+++ b/net/minecraft/commands/arguments/ResourceOrTagArgument.java
@@ -150,7 +150,7 @@
 
         @Override
         public <E> Optional<ResourceOrTagArgument.Result<E>> cast(ResourceKey<? extends Registry<E>> resourceKey) {
-            return this.value.key().isFor(resourceKey) ? Optional.of(this) : Optional.empty();
+            return this.value.key().isFor(resourceKey) ? Optional.of((ResourceOrTagArgument.Result<E>) this) : Optional.empty();
         }
 
         @Override
@@ -180,7 +180,7 @@
 
         @Override
         public <E> Optional<ResourceOrTagArgument.Result<E>> cast(ResourceKey<? extends Registry<E>> resourceKey) {
-            return this.tag.key().isFor(resourceKey) ? Optional.of(this) : Optional.empty();
+            return this.tag.key().isFor(resourceKey) ? Optional.of((ResourceOrTagArgument.Result<E>) this) : Optional.empty();
         }
 
         @Override
