--- a/net/minecraft/util/datafix/DataFixers.java
+++ b/net/minecraft/util/datafix/DataFixers.java
@@ -1008,7 +1008,7 @@
                 References.ENTITY,
                 "minecraft:cat",
                 "CatType",
-                Util.make(new Int2ObjectOpenHashMap(), int2ObjectOpenHashMap -> {
+                Util.make(new Int2ObjectOpenHashMap<String>(), int2ObjectOpenHashMap -> {
                     int2ObjectOpenHashMap.defaultReturnValue("minecraft:tabby");
                     int2ObjectOpenHashMap.put(0, "minecraft:tabby");
                     int2ObjectOpenHashMap.put(1, "minecraft:black");
@@ -1050,7 +1050,7 @@
                 References.ENTITY,
                 "minecraft:frog",
                 "Variant",
-                Util.make(new Int2ObjectOpenHashMap(), int2ObjectOpenHashMap -> {
+                Util.make(new Int2ObjectOpenHashMap<String>(), int2ObjectOpenHashMap -> {
                     int2ObjectOpenHashMap.put(0, "minecraft:temperate");
                     int2ObjectOpenHashMap.put(1, "minecraft:warm");
                     int2ObjectOpenHashMap.put(2, "minecraft:cold");
