--- a/net/minecraft/resources/RegistryFileCodec.java
+++ b/net/minecraft/resources/RegistryFileCodec.java
@@ -69,7 +69,7 @@
                     return holderGetter.get(resourceKey)
                         .map(DataResult::success)
                         .orElseGet(() -> DataResult.error(() -> "Failed to get element " + resourceKey))
-                        .<Pair<? super Holder.Reference<E>, T>>map(reference -> Pair.of(reference, pair.getSecond()))
+                        .map(reference -> Pair.of((Holder<E>) reference, pair.getSecond()))
                         .setLifecycle(Lifecycle.stable());
                 }
             }
