--- a/net/minecraft/resources/RegistryFixedCodec.java
+++ b/net/minecraft/resources/RegistryFixedCodec.java
@@ -56,7 +56,7 @@
                                 .get(ResourceKey.create(this.registryKey, resourceLocation))
                                 .map(DataResult::success)
                                 .orElseGet(() -> DataResult.error(() -> "Failed to get element " + resourceLocation))
-                                .<Pair<? super Holder.Reference<E>, T>>map(reference -> Pair.of(reference, (T)pair.getSecond()))
+                                .map(reference -> Pair.of((Holder<E>) reference, (T)pair.getSecond()))
                                 .setLifecycle(Lifecycle.stable());
                         }
                     );
