--- a/net/minecraft/util/ClassInstanceMultiMap.java
+++ b/net/minecraft/util/ClassInstanceMultiMap.java
@@ -61,7 +61,7 @@
         } else {
             List<? extends T> list = this.byClass
                 .computeIfAbsent(type, clazz -> this.allInstances.stream().filter(clazz::isInstance).collect(Collectors.toList()));
-            return Collections.unmodifiableCollection(list);
+            return (Collection<S>) Collections.unmodifiableCollection(list);
         }
     }
 
