--- a/net/minecraft/world/entity/player/StackedContents.java
+++ b/net/minecraft/world/entity/player/StackedContents.java
@@ -190,29 +190,29 @@
                     this.visit(false, i1);
 
                     while(!this.path.isEmpty()) {
-                        int size = this.path.size();
-                        boolean flag = (size & 1) == 1;
-                        int _int = this.path.getInt(size - 1);
+                        int i2 = this.path.size();
+                        boolean flag = (i2 & 1) == 1;
+                        int _int = this.path.getInt(i2 - 1);
                         if (!flag && !this.isSatisfied(_int)) {
                             break;
                         }
 
-                        int i2 = flag ? this.ingredientCount : i;
-                        int i3 = 0;
+                        int i3 = flag ? this.ingredientCount : i;
+                        int i4 = 0;
 
                         while(true) {
-                            if (i3 < i2) {
-                                if (this.hasVisited(flag, i3) || !this.hasConnection(flag, _int, i3) || !this.hasResidual(flag, _int, i3)) {
-                                    ++i3;
+                            if (i4 < i3) {
+                                if (this.hasVisited(flag, i4) || !this.hasConnection(flag, _int, i4) || !this.hasResidual(flag, _int, i4)) {
+                                    ++i4;
                                     continue;
                                 }
 
-                                this.visit(flag, i3);
+                                this.visit(flag, i4);
                             }
 
-                            i3 = this.path.size();
-                            if (i3 == size) {
-                                this.path.removeInt(i3 - 1);
+                            i4 = this.path.size();
+                            if (i4 == i2) {
+                                this.path.removeInt(i4 - 1);
                             }
                             break;
                         }
