--- a/net/minecraft/util/datafix/fixes/AddNewChoices.java
+++ b/net/minecraft/util/datafix/fixes/AddNewChoices.java
@@ -4,6 +4,7 @@
 import com.mojang.datafixers.TypeRewriteRule;
 import com.mojang.datafixers.DSL.TypeReference;
 import com.mojang.datafixers.schemas.Schema;
+import com.mojang.datafixers.types.templates.TaggedChoice;
 import com.mojang.datafixers.types.templates.TaggedChoice.TaggedChoiceType;
 import java.util.Locale;
 
@@ -28,8 +29,8 @@
         if (type.getKeyType() != newType.getKeyType()) {
             throw new IllegalStateException("Could not inject: key type is not the same");
         } else {
-            return this.fixTypeEverywhere(name, type, newType, dynamicOps -> pair -> {
-                    if (!newType.hasType(pair.getFirst())) {
+            return this.fixTypeEverywhere(name, type, (TaggedChoice.TaggedChoiceType<K>) newType, dynamicOps -> pair -> {
+                    if (!((TaggedChoice.TaggedChoiceType<K>) newType).hasType(pair.getFirst())) {
                         throw new IllegalArgumentException(String.format(Locale.ROOT, "Unknown type %s in %s ", pair.getFirst(), this.type));
                     } else {
                         return pair;
