--- a/net/minecraft/world/entity/monster/Shulker.java
+++ b/net/minecraft/world/entity/monster/Shulker.java
@@ -457,8 +457,8 @@
         Vec3 vec3 = this.position();
         AABB boundingBox = this.getBoundingBox();
         if (!this.isClosed() && this.teleportSomewhere()) {
-            int size = this.level().getEntities(EntityType.SHULKER, boundingBox.inflate(8.0), Entity::isAlive).size();
-            float f = (float)(size - 1) / 5.0F;
+            int i = this.level().getEntities(EntityType.SHULKER, boundingBox.inflate(8.0), Entity::isAlive).size();
+            float f = (float)(i - 1) / 5.0F;
             if (!(this.level().random.nextFloat() < f)) {
                 Shulker shulker = EntityType.SHULKER.create(this.level());
                 if (shulker != null) {
