--- a/net/minecraft/world/entity/monster/Vex.java
+++ b/net/minecraft/world/entity/monster/Vex.java
@@ -320,20 +320,20 @@
         public void tick() {
             if (this.operation == MoveControl.Operation.MOVE_TO) {
                 Vec3 vec3 = new Vec3(this.wantedX - Vex.this.getX(), this.wantedY - Vex.this.getY(), this.wantedZ - Vex.this.getZ());
-                double len = vec3.length();
-                if (len < Vex.this.getBoundingBox().getSize()) {
+                double d = vec3.length();
+                if (d < Vex.this.getBoundingBox().getSize()) {
                     this.operation = MoveControl.Operation.WAIT;
                     Vex.this.setDeltaMovement(Vex.this.getDeltaMovement().scale(0.5));
                 } else {
-                    Vex.this.setDeltaMovement(Vex.this.getDeltaMovement().add(vec3.scale(this.speedModifier * 0.05 / len)));
+                    Vex.this.setDeltaMovement(Vex.this.getDeltaMovement().add(vec3.scale(this.speedModifier * 0.05 / d)));
                     if (Vex.this.getTarget() == null) {
                         Vec3 deltaMovement = Vex.this.getDeltaMovement();
                         Vex.this.setYRot(-((float)Mth.atan2(deltaMovement.x, deltaMovement.z)) * (180.0F / (float)Math.PI));
                         Vex.this.yBodyRot = Vex.this.getYRot();
                     } else {
                         double deltaMovement1 = Vex.this.getTarget().getX() - Vex.this.getX();
-                        double d = Vex.this.getTarget().getZ() - Vex.this.getZ();
-                        Vex.this.setYRot(-((float)Mth.atan2(deltaMovement1, d)) * (180.0F / (float)Math.PI));
+                        double d1 = Vex.this.getTarget().getZ() - Vex.this.getZ();
+                        Vex.this.setYRot(-((float)Mth.atan2(deltaMovement1, d1)) * (180.0F / (float)Math.PI));
                         Vex.this.yBodyRot = Vex.this.getYRot();
                     }
                 }
