--- a/net/minecraft/world/inventory/RecipeBookMenu.java
+++ b/net/minecraft/world/inventory/RecipeBookMenu.java
@@ -12,7 +12,7 @@
     }
 
     public void handlePlacement(boolean craftAll, Recipe<?> recipe, ServerPlayer serverPlayer) {
-        new ServerPlaceRecipe<>(this).recipeClicked(serverPlayer, recipe, craftAll);
+        new ServerPlaceRecipe<>(this).recipeClicked(serverPlayer, (Recipe<C>) recipe, craftAll);
     }
 
     public abstract void fillCraftSlotsStackedContents(StackedContents finder);
