--- a/net/minecraft/world/entity/ai/Brain.java
+++ b/net/minecraft/world/entity/ai/Brain.java
@@ -198,14 +198,14 @@
         if (optional == null) {
             throw new IllegalStateException("Unregistered memory fetched: " + memoryModuleType);
         } else {
-            return optional.map(ExpirableValue::getValue);
+            return (Optional<U>) optional.map(ExpirableValue::getValue);
         }
     }
 
     @Nullable
     public <U> Optional<U> getMemoryInternal(MemoryModuleType<U> memoryModuleType) {
         Optional<? extends ExpirableValue<?>> optional = this.memories.get(memoryModuleType);
-        return optional == null ? null : optional.map(ExpirableValue::getValue);
+        return optional == null ? null : (Optional<U>) optional.map(ExpirableValue::getValue);
     }
 
     public <U> long getTimeUntilExpiry(MemoryModuleType<U> memoryModuleType) {
@@ -499,7 +499,7 @@
         private final Optional<? extends ExpirableValue<U>> value;
 
         static <U> Brain.MemoryValue<U> createUnchecked(MemoryModuleType<U> memoryModuleType, Optional<? extends ExpirableValue<?>> data) {
-            return new Brain.MemoryValue<>(memoryModuleType, data);
+            return new Brain.MemoryValue<>(memoryModuleType, (Optional<ExpirableValue<U>>) data);
         }
 
         MemoryValue(MemoryModuleType<U> type, Optional<? extends ExpirableValue<U>> optional) {
