--- a/net/minecraft/world/item/component/PiercingWeapon.java
+++ b/net/minecraft/world/item/component/PiercingWeapon.java
@@ -78,7 +_,7 @@
         for (EntityHitResult entityHitResult : ProjectileUtil.getHitEntitiesAlong(
                 entity, attackRange, entity1 -> canHitEntity(entity, entity1), ClipContext.Block.COLLIDER
             )
-            .map(blockHitResult -> List.of(), collection -> (Collection)collection)) {
+            .<Collection<EntityHitResult>>map(blockHitResult -> List.of(), collection -> collection)) {
             flag |= entity.stabAttack(slot, entityHitResult.getEntity(), f, true, this.dealsKnockback, this.dismounts);
         }
 
