--- a/net/minecraft/util/datafix/fixes/BlockEntityRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockEntityRenameFix.java
@@ -18,8 +18,8 @@
 
     @Override
     public TypeRewriteRule makeRule() {
-        TaggedChoiceType<String> taggedChoiceType = this.getInputSchema().findChoiceType(References.BLOCK_ENTITY);
-        TaggedChoiceType<String> taggedChoiceType1 = this.getOutputSchema().findChoiceType(References.BLOCK_ENTITY);
+        TaggedChoiceType<String> taggedChoiceType = (TaggedChoiceType<String>) this.getInputSchema().findChoiceType(References.BLOCK_ENTITY);
+        TaggedChoiceType<String> taggedChoiceType1 = (TaggedChoiceType<String>) this.getOutputSchema().findChoiceType(References.BLOCK_ENTITY);
         return this.fixTypeEverywhere(this.name, taggedChoiceType, taggedChoiceType1, ops -> pair -> pair.mapFirst(this.nameChangeLookup));
     }
 
