--- a/net/minecraft/util/datafix/fixes/BlockRenameFixWithJigsaw.java
+++ b/net/minecraft/util/datafix/fixes/BlockRenameFixWithJigsaw.java
@@ -34,18 +34,18 @@
                             dynamic -> dynamic.update("final_state", dynamic1 -> DataFixUtils.orElse(dynamic1.asString().result().map(string1 -> {
                                         int index = string1.indexOf(91);
                                         int index1 = string1.indexOf(123);
-                                        int len = string1.length();
+                                        int i = string1.length();
                                         if (index > 0) {
-                                            len = Math.min(len, index);
+                                            i = Math.min(i, index);
                                         }
             
                                         if (index1 > 0) {
-                                            len = Math.min(len, index1);
+                                            i = Math.min(i, index1);
                                         }
             
-                                        String sub = string1.substring(0, len);
+                                        String sub = string1.substring(0, i);
                                         String string2 = this.fixBlock(sub);
-                                        return string2 + string1.substring(len);
+                                        return string2 + string1.substring(i);
                                     }).map(dynamic::createString), dynamic1))
                         )
                 )
