--- a/net/minecraft/util/datafix/fixes/ChunkHeightAndBiomeFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkHeightAndBiomeFix.java
@@ -459,16 +459,16 @@
     private static Dynamic<?> padPaletteEntries(Dynamic<?> dynamic, Dynamic<?> dynamic1, List<Dynamic<?>> list) {
         long l = dynamic1.asLongStream().count() * 64L;
         long l1 = l / 4096L;
-        int size = list.size();
-        int i = ceillog2(size);
-        if (l1 <= (long)i) {
+        int i = list.size();
+        int i1 = ceillog2(i);
+        if (l1 <= (long)i1) {
             return dynamic;
         } else {
             Dynamic<?> dynamic2 = dynamic.createMap(ImmutableMap.of(dynamic.createString("Name"), dynamic.createString("minecraft:air")));
-            int i1 = (1 << (int)(l1 - 1L)) + 1;
-            int i2 = i1 - size;
+            int i2 = (1 << (int)(l1 - 1L)) + 1;
+            int i3 = i2 - i;
 
-            for(int i3 = 0; i3 < i2; ++i3) {
+            for(int i4 = 0; i4 < i3; ++i4) {
                 list.add(dynamic2);
             }
 
