--- a/net/minecraft/util/datafix/fixes/ChunkHeightAndBiomeFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkHeightAndBiomeFix.java
@@ -221,12 +221,14 @@
             heightAlreadyUpdated.setValue(true);
 
             for(int i1 = 0; i1 < 24; ++i1) {
-                dynamics[i1] = makeBiomeContainer(level, sectionY -> getOldBiome(ints, i1 * 64 + sectionY));
+                final int i1final = i1;
+                dynamics[i1] = makeBiomeContainer(level, sectionY -> getOldBiome(ints, i1final * 64 + sectionY));
             }
         } else if (ints != null && ints.length == 1024) {
             for(int i1 = 0; i1 < 16; ++i1) {
                 int i2 = i1 - i;
-                dynamics[i2] = makeBiomeContainer(level, sectionY -> getOldBiome(ints, i1 * 64 + sectionY));
+                final int i1final = i1;
+                dynamics[i2] = makeBiomeContainer(level, sectionY -> getOldBiome(ints, i1final * 64 + sectionY));
             }
 
             if (overworld) {
