--- a/net/minecraft/util/datafix/fixes/ChunkProtoTickListFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkProtoTickListFix.java
@@ -113,7 +113,7 @@
                                 var18 = var18.set(opticFinder6, ((Pair)optional.get()).getFirst());
                             }
         
-                            return var18.update(
+                            return ((Typed<?>) var18).update(
                                 DSL.remainderFinder(), dynamic4 -> dynamic4.remove("ToBeTicked").remove("LiquidsToBeTicked").set("fluid_ticks", dynamic1)
                             );
                         } else {
@@ -185,7 +185,7 @@
         int i6 = i3 >>> 8 & 15;
         String string = function.apply(supplier != null ? supplier.get().get(i4, i5, i6) : null);
         return dynamic.createMap(
-            ImmutableMap.builder()
+            ImmutableMap.<Dynamic<?>, Dynamic<?>>builder()
                 .put(dynamic.createString("i"), dynamic.createString(string))
                 .put(dynamic.createString("x"), dynamic.createInt(i * 16 + i4))
                 .put(dynamic.createString("y"), dynamic.createInt(i1 * 16 + i5))
