--- a/net/minecraft/world/level/block/Block.java
+++ b/net/minecraft/world/level/block/Block.java
@@ -524,7 +_,7 @@
     }
 
     private static <S extends StateHolder<?, S>, T extends Comparable<T>> S setValueHelper(S stateHolder, Property<T> property, Object object) {
-        return stateHolder.setValue(property, (Comparable)object);
+        return stateHolder.setValue(property, (T) object);
     }
 
     @Deprecated
