--- a/net/minecraft/advancements/Criterion.java
+++ b/net/minecraft/advancements/Criterion.java
@@ -18,7 +18,7 @@
         if (criterion == null) {
             throw new JsonSyntaxException("Invalid criterion trigger: " + resourceLocation);
         } else {
-            return criterionFromJson(json, context, (CriterionTrigger<T>)criterion);
+            return criterionFromJson(json, context, criterion);
         }
     }
 
