--- a/net/minecraft/server/dedicated/Settings.java
+++ b/net/minecraft/server/dedicated/Settings.java
@@ -119,7 +119,7 @@
         String stringRaw = this.getStringRaw(key);
         V object = MoreObjects.firstNonNull(stringRaw != null ? mapper.apply(stringRaw) : null, value);
         this.properties.put(key, toString.apply(object));
-        return new Settings.MutableValue<>(key, object, toString);
+        return new Settings<T>.MutableValue<>(key, object, toString);
     }
 
     protected <V> V get(String key, Function<String, V> mapper, UnaryOperator<V> unaryOperator, Function<V, String> toString, V value) {
