--- a/net/minecraft/server/dedicated/Settings.java
+++ b/net/minecraft/server/dedicated/Settings.java
@@ -119,7 +119,7 @@
         String s = this.getStringRaw(key);
         V v = MoreObjects.firstNonNull(s != null ? parser.apply(s) : null, fallback);
         this.properties.put(key, stringifier.apply(v));
-        return new Settings.MutableValue<>(key, v, stringifier);
+        return new Settings<T>.MutableValue<>(key, v, stringifier);
     }
 
     protected <V> V get(String key, Function<String, V> parser, UnaryOperator<V> parsedTransformer, Function<V, String> stringifier, V fallback) {
