--- a/net/minecraft/world/entity/ai/Brain.java
+++ b/net/minecraft/world/entity/ai/Brain.java
@@ -157,13 +_,13 @@
         if (expirableValue == null) {
             throw new IllegalStateException("Unregistered memory fetched: " + type);
         } else {
-            return expirableValue.map(ExpirableValue::getValue);
+            return (Optional<U>) expirableValue.map(ExpirableValue::getValue);
         }
     }
 
     public <U> @Nullable Optional<U> getMemoryInternal(final MemoryModuleType<U> type) {
         Optional<? extends ExpirableValue<?>> expirableValue = this.memories.get(type);
-        return expirableValue == null ? null : expirableValue.map(ExpirableValue::getValue);
+        return expirableValue == null ? null : (Optional<U>) expirableValue.map(ExpirableValue::getValue);
     }
 
     public <U> long getTimeUntilExpiry(final MemoryModuleType<U> type) {
