--- a/net/minecraft/world/level/storage/loot/LootDataType.java
+++ b/net/minecraft/world/level/storage/loot/LootDataType.java
@@ -80,12 +80,12 @@
     private static <T> BiFunction<Gson, String, BiFunction<ResourceLocation, JsonElement, Optional<T>>> createSingleOrMultipleDeserialiser(
         Class<T> clazz, Function<T[], T> function
     ) {
-        Class<T[]> clazz1 = clazz.arrayType();
+        Class<T[]> clazz1 = (Class<T[]>) clazz.arrayType();
         return (gson, string) -> (resourceLocation, jsonElement) -> {
                 try {
                     if (jsonElement.isJsonArray()) {
                         T[] objects = (T[])((Object[])gson.fromJson(jsonElement, clazz1));
-                        return Optional.of(function.apply((T)objects));
+                        return Optional.of(function.apply(objects));
                     } else {
                         return Optional.of(gson.fromJson(jsonElement, clazz));
                     }
