--- a/net/minecraft/util/datafix/fixes/BlockEntityUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockEntityUUIDFix.java
@@ -21,7 +21,7 @@
         return dynamic.get("Owner")
             .get()
             .map(dynamicx -> replaceUUIDString(dynamicx, "Id", "Id").orElse(dynamicx))
-            .map(dynamic2 -> dynamic.remove("Owner").set("SkullOwner", dynamic2))
+            .<Dynamic<?>>map(dynamic2 -> dynamic.remove("Owner").set("SkullOwner", dynamic2))
             .result()
             .orElse(dynamic);
     }
