--- a/net/minecraft/util/datafix/fixes/ChunkBedBlockEntityInjecterFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkBedBlockEntityInjecterFix.java
@@ -44,7 +44,7 @@
             this.fixTypeEverywhere(
                 "InjectBedBlockEntityType",
                 this.getInputSchema().findChoiceType(References.BLOCK_ENTITY),
-                this.getOutputSchema().findChoiceType(References.BLOCK_ENTITY),
+                (com.mojang.datafixers.types.templates.TaggedChoice.TaggedChoiceType) this.getOutputSchema().findChoiceType(References.BLOCK_ENTITY),
                 dynamicOps -> pair -> pair
             ),
             this.fixTypeEverywhereTyped(
