--- a/net/minecraft/util/datafix/fixes/OptionsKeyTranslationFix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsKeyTranslationFix.java
@@ -22,12 +22,12 @@
                             if (((Dynamic)entry.getKey()).asString("").startsWith("key_")) {
                                 String string = ((Dynamic)entry.getValue()).asString("");
                                 if (!string.startsWith("key.mouse") && !string.startsWith("scancode.")) {
-                                    return Pair.of((Dynamic)entry.getKey(), dynamic.createString("key.keyboard." + string.substring("key.".length())));
+                                    return Pair.of(entry.getKey(), dynamic.createString("key.keyboard." + string.substring("key.".length())));
                                 }
                             }
     
-                            return Pair.of((Dynamic)entry.getKey(), (Dynamic)entry.getValue());
-                        }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)))).result().orElse(dynamic))
+                            return Pair.of(entry.getKey(), entry.getValue());
+                        }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)))).result().orElse((com.mojang.serialization.Dynamic) dynamic))
         );
     }
 }
