--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -981,7 +_,7 @@
 
         Entity entity = source.getEntity();
         return !(entity instanceof Player player && !this.canHarmPlayer(player))
-            && !(entity instanceof AbstractArrow arrow && arrow.getOwner() instanceof Player player && !this.canHarmPlayer(player))
+            && !(entity instanceof AbstractArrow arrow && arrow.getOwner() instanceof Player playerx && !this.canHarmPlayer(playerx))
             && super.hurtServer(level, source, damage);
     }
 
