--- a/net/minecraft/world/level/levelgen/feature/BlockColumnFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/BlockColumnFeature.java
@@ -17,36 +17,36 @@
         WorldGenLevel worldGenLevel = context.level();
         BlockColumnConfiguration blockColumnConfiguration = context.config();
         RandomSource randomSource = context.random();
-        int size = blockColumnConfiguration.layers().size();
-        int[] ints = new int[size];
-        int i = 0;
+        int i = blockColumnConfiguration.layers().size();
+        int[] ints = new int[i];
+        int i1 = 0;
 
-        for(int i1 = 0; i1 < size; ++i1) {
-            ints[i1] = blockColumnConfiguration.layers().get(i1).height().sample(randomSource);
-            i += ints[i1];
+        for(int i2 = 0; i2 < i; ++i2) {
+            ints[i2] = blockColumnConfiguration.layers().get(i2).height().sample(randomSource);
+            i1 += ints[i2];
         }
 
-        if (i == 0) {
+        if (i1 == 0) {
             return false;
         } else {
             BlockPos.MutableBlockPos mutableBlockPos = context.origin().mutable();
             BlockPos.MutableBlockPos mutableBlockPos1 = mutableBlockPos.mutable().move(blockColumnConfiguration.direction());
 
-            for(int i2 = 0; i2 < i; ++i2) {
+            for(int i3 = 0; i3 < i1; ++i3) {
                 if (!blockColumnConfiguration.allowedPlacement().test(worldGenLevel, mutableBlockPos1)) {
-                    truncate(ints, i, i2, blockColumnConfiguration.prioritizeTip());
+                    truncate(ints, i1, i3, blockColumnConfiguration.prioritizeTip());
                     break;
                 }
 
                 mutableBlockPos1.move(blockColumnConfiguration.direction());
             }
 
-            for(int i2 = 0; i2 < size; ++i2) {
-                int i3 = ints[i2];
-                if (i3 != 0) {
-                    BlockColumnConfiguration.Layer layer = blockColumnConfiguration.layers().get(i2);
+            for(int i3 = 0; i3 < i; ++i3) {
+                int i4 = ints[i3];
+                if (i4 != 0) {
+                    BlockColumnConfiguration.Layer layer = blockColumnConfiguration.layers().get(i3);
 
-                    for(int i4 = 0; i4 < i3; ++i4) {
+                    for(int i5 = 0; i5 < i4; ++i5) {
                         worldGenLevel.setBlock(mutableBlockPos, layer.state().getState(randomSource, mutableBlockPos), Block.UPDATE_CLIENTS);
                         mutableBlockPos.move(blockColumnConfiguration.direction());
                     }
