--- a/net/minecraft/world/level/block/BaseEntityBlock.java
+++ b/net/minecraft/world/level/block/BaseEntityBlock.java
@@ -35,6 +_,6 @@
     protected static <E extends BlockEntity, A extends BlockEntity> @Nullable BlockEntityTicker<A> createTickerHelper(
         BlockEntityType<A> actualType, BlockEntityType<E> expectedType, BlockEntityTicker<? super E> ticker
     ) {
-        return expectedType == actualType ? ticker : null;
+        return expectedType == actualType ? (BlockEntityTicker<A>) ticker : null;
     }
 }
