--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -473,6 +_,7 @@
                 }
 
                 vehicle.move(MoverType.PLAYER, new Vec3(xDist, yDist, zDist));
+                double oyDist = yDist;
                 xDist = targetX - vehicle.getX();
                 yDist = targetY - vehicle.getY();
                 if (yDist > -0.5 || yDist < 0.5) {
@@ -501,7 +_,7 @@
                 vehicle.setOnGroundWithMovement(packet.onGround(), clientDeltaMovement);
                 vehicle.doCheckFallDamage(clientDeltaMovement.x, clientDeltaMovement.y, clientDeltaMovement.z, packet.onGround());
                 this.player.checkMovementStatistics(clientDeltaMovement.x, clientDeltaMovement.y, clientDeltaMovement.z);
-                this.clientVehicleIsFloating = yDist >= -0.03125
+                this.clientVehicleIsFloating = oyDist >= -0.03125
                     && !vehicleRestsOnSomething
                     && !this.server.allowFlight()
                     && !vehicle.isFlyingVehicle()
@@ -998,7 +_,7 @@
         if (carried.has(DataComponents.WRITABLE_BOOK_CONTENT)) {
             ItemStack writtenBook = carried.transmuteCopy(Items.WRITTEN_BOOK);
             writtenBook.remove(DataComponents.WRITABLE_BOOK_CONTENT);
-            List<Filterable<Component>> pages = contents.stream().map(page -> this.filterableFromOutgoing(page).map(Component::literal)).toList();
+            List<Filterable<Component>> pages = contents.stream().map(page -> this.filterableFromOutgoing(page).<Component>map(Component::literal)).toList();
             writtenBook.set(
                 DataComponents.WRITTEN_BOOK_CONTENT, new WrittenBookContent(this.filterableFromOutgoing(title), this.player.getPlainTextName(), 0, pages, true)
             );
@@ -1120,6 +_,7 @@
 
                                 boolean playerStandsOnSomething = this.player.verticalCollisionBelow;
                                 this.player.move(MoverType.PLAYER, new Vec3(xDist, yDist, zDist));
+                                double oyDist = yDist;
                                 xDist = targetX - this.player.getX();
                                 yDist = targetY - this.player.getY();
                                 if (yDist > -0.5 || yDist < 0.5) {
@@ -1145,7 +_,7 @@
                                         && !this.isEntityCollidingWithAnythingNew(level, this.player, oldAABB, targetX, targetY, targetZ)) {
                                     this.player.absSnapTo(targetX, targetY, targetZ, targetYRot, targetXRot);
                                     boolean isAutoSpinAttack = this.player.isAutoSpinAttack();
-                                    this.clientIsFloating = yDist >= -0.03125
+                                    this.clientIsFloating = oyDist >= -0.03125
                                         && !playerStandsOnSomething
                                         && !this.player.isSpectator()
                                         && !this.server.allowFlight()
