--- a/net/minecraft/util/datafix/fixes/ChunkProtoTickListFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkProtoTickListFix.java
@@ -181,7 +_,7 @@
         int i3 = i >>> 8 & 15;
         String string = function.apply(supplier != null ? supplier.get().get(i1, i2, i3) : null);
         return dynamic.createMap(
-            ImmutableMap.builder()
+            ImmutableMap.<Dynamic<?>, Dynamic<?>>builder()
                 .put(dynamic.createString("i"), dynamic.createString(string))
                 .put(dynamic.createString("x"), dynamic.createInt(x * 16 + i1))
                 .put(dynamic.createString("y"), dynamic.createInt(y * 16 + i2))
