--- a/net/minecraft/network/syncher/EntityDataSerializer.java
+++ b/net/minecraft/network/syncher/EntityDataSerializer.java
@@ -19,12 +19,12 @@
         return new EntityDataSerializer.ForValueType<T>() {
             @Override
             public void write(FriendlyByteBuf buffer, T value) {
-                valueWriter.accept((T)buffer, value);
+                valueWriter.accept(buffer, value);
             }
 
             @Override
             public T read(FriendlyByteBuf buffer) {
-                return valueReader.apply((T)buffer);
+                return valueReader.apply(buffer);
             }
         };
     }
