--- a/net/minecraft/core/registries/BuiltInRegistries.java
+++ b/net/minecraft/core/registries/BuiltInRegistries.java
@@ -298,7 +298,7 @@
     ) {
         ResourceLocation resourceLocation = key.location();
         LOADERS.put(resourceLocation, () -> bootstrap.run(registry));
-        WRITABLE_REGISTRY.register((ResourceKey<WritableRegistry<?>>)key, registry, lifecycle);
+        WRITABLE_REGISTRY.register((ResourceKey)key, registry, lifecycle);
         return registry;
     }
 
