--- a/net/minecraft/network/protocol/game/ClientboundUpdateEnabledFeaturesPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundUpdateEnabledFeaturesPacket.java
@@ -8,7 +8,7 @@
 
 public record ClientboundUpdateEnabledFeaturesPacket(Set<ResourceLocation> features) implements Packet<ClientGamePacketListener> {
     public ClientboundUpdateEnabledFeaturesPacket(FriendlyByteBuf buf) {
-        this(buf.readCollection(HashSet::new, FriendlyByteBuf::readResourceLocation));
+        this(buf.<ResourceLocation, Set<ResourceLocation>>readCollection(HashSet::new, FriendlyByteBuf::readResourceLocation));
     }
 
     @Override
