--- a/net/minecraft/server/commands/ForceLoadCommand.java
+++ b/net/minecraft/server/commands/ForceLoadCommand.java
@@ -114,19 +114,19 @@
         ServerLevel level = source.getLevel();
         ResourceKey<Level> resourceKey = level.dimension();
         LongSet forcedChunks = level.getForcedChunks();
-        int size = forcedChunks.size();
-        if (size > 0) {
+        int i = forcedChunks.size();
+        if (i > 0) {
             String string = Joiner.on(", ").join(forcedChunks.stream().sorted().map(ChunkPos::new).map(ChunkPos::toString).iterator());
-            if (size == 1) {
+            if (i == 1) {
                 source.sendSuccess(() -> Component.translatable("commands.forceload.list.single", resourceKey.location(), string), false);
             } else {
-                source.sendSuccess(() -> Component.translatable("commands.forceload.list.multiple", size, resourceKey.location(), string), false);
+                source.sendSuccess(() -> Component.translatable("commands.forceload.list.multiple", i, resourceKey.location(), string), false);
             }
         } else {
             source.sendFailure(Component.translatable("commands.forceload.added.none", resourceKey.location()));
         }
 
-        return size;
+        return i;
     }
 
     private static int removeAll(CommandSourceStack source) {
