--- a/net/minecraft/world/level/levelgen/heightproviders/ConstantHeight.java
+++ b/net/minecraft/world/level/levelgen/heightproviders/ConstantHeight.java
@@ -11,7 +11,7 @@
     public static final ConstantHeight ZERO = new ConstantHeight(VerticalAnchor.absolute(0));
     public static final Codec<ConstantHeight> CODEC = Codec.either(
             VerticalAnchor.CODEC,
-            RecordCodecBuilder.create(
+            RecordCodecBuilder.<ConstantHeight>create(
                 instance -> instance.group(VerticalAnchor.CODEC.fieldOf("value").forGetter(constantHeight -> constantHeight.value))
                         .apply(instance, ConstantHeight::new)
             )
