--- a/net/minecraft/world/level/chunk/LevelChunk.java
+++ b/net/minecraft/world/level/chunk/LevelChunk.java
@@ -617,7 +617,7 @@
 
     private <T extends BlockEntity> void updateBlockEntityTicker(T blockEntity) {
         BlockState blockState = blockEntity.getBlockState();
-        BlockEntityTicker<T> ticker = blockState.getTicker(this.level, blockEntity.getType());
+        BlockEntityTicker<T> ticker = (BlockEntityTicker<T>) blockState.getTicker(this.level, blockEntity.getType());
         if (ticker == null) {
             this.removeBlockEntityTicker(blockEntity.getBlockPos());
         } else {
