--- a/net/minecraft/world/level/chunk/LevelChunk.java
+++ b/net/minecraft/world/level/chunk/LevelChunk.java
@@ -617,7 +617,7 @@
 
     private <T extends BlockEntity> void updateBlockEntityTicker(T blockEntity) {
         BlockState blockstate = blockEntity.getBlockState();
-        BlockEntityTicker<T> blockentityticker = blockstate.getTicker(this.level, blockEntity.getType());
+        BlockEntityTicker<T> blockentityticker = (BlockEntityTicker<T>) blockstate.getTicker(this.level, blockEntity.getType());
         if (blockentityticker == null) {
             this.removeBlockEntityTicker(blockEntity.getBlockPos());
         } else {
