--- a/net/minecraft/world/entity/projectile/FishingHook.java
+++ b/net/minecraft/world/entity/projectile/FishingHook.java
@@ -91,11 +91,9 @@
         double d1 = thrower.getZ() - (double)f * 0.3;
         this.moveTo(d, eyeY, d1, yRot, xRot);
         Vec3 vec3 = new Vec3((double)(-f1), (double)Mth.clamp(-(f3 / f2), -5.0F, 5.0F), (double)(-f));
-        double len = vec3.length();
+        double d2 = vec3.length();
         vec3 = vec3.multiply(
-            0.6 / len + this.random.triangle(0.5, 0.0103365),
-            0.6 / len + this.random.triangle(0.5, 0.0103365),
-            0.6 / len + this.random.triangle(0.5, 0.0103365)
+            0.6 / d2 + this.random.triangle(0.5, 0.0103365), 0.6 / d2 + this.random.triangle(0.5, 0.0103365), 0.6 / d2 + this.random.triangle(0.5, 0.0103365)
         );
         this.setDeltaMovement(vec3);
         this.setYRot((float)(Mth.atan2(vec3.x, vec3.z) * 180.0F / (float)Math.PI));
