--- a/net/minecraft/world/entity/ai/behavior/RandomStroll.java
+++ b/net/minecraft/world/entity/ai/behavior/RandomStroll.java
@@ -43,7 +43,7 @@
 
     private static OneShot<PathfinderMob> strollFlyOrSwim(float speedModifier, Function<PathfinderMob, Vec3> target, Predicate<PathfinderMob> canStroll) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor>group(instance.absent(MemoryModuleType.WALK_TARGET)).apply(instance, walkTarget -> (level, mob, gameTime) -> {
+            instance -> instance.group(instance.absent(MemoryModuleType.WALK_TARGET)).apply(instance, walkTarget -> (level, mob, gameTime) -> {
                         if (!canStroll.test(mob)) {
                             return false;
                         } else {
