--- a/net/minecraft/world/entity/EntityReference.java
+++ b/net/minecraft/world/entity/EntityReference.java
@@ -116,7 +_,7 @@
     }
 
     public static <StoredEntityType extends UniquelyIdentifyable> @Nullable EntityReference<StoredEntityType> read(ValueInput input, String key) {
-        return input.read(key, codec()).orElse(null);
+        return input.read(key, EntityReference.<StoredEntityType>codec()).orElse(null);
     }
 
     public static <StoredEntityType extends UniquelyIdentifyable> @Nullable EntityReference<StoredEntityType> readWithOldOwnerConversion(
@@ -127,7 +_,7 @@
             ? of(optional.get())
             : input.getString(key)
                 .map(string -> OldUsersConverter.convertMobOwnerIfNecessary(level.getServer(), string))
-                .map(EntityReference::new)
+                .map(EntityReference<StoredEntityType>::new)
                 .orElse(null);
     }
 
