--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -1184,6 +_,7 @@
             }
 
             ItemStack useItem = this.getUseItem();
+            float originAmount = amount;
             float f1 = this.applyItemBlocking(level, damageSource, amount);
             amount -= f1;
             boolean flag = f1 > 0.0F;
@@ -1275,14 +_,14 @@
             }
 
             if (this instanceof ServerPlayer serverPlayer) {
-                CriteriaTriggers.ENTITY_HURT_PLAYER.trigger(serverPlayer, damageSource, amount, amount, flag);
+                CriteriaTriggers.ENTITY_HURT_PLAYER.trigger(serverPlayer, damageSource, originAmount, amount, flag);
                 if (f1 > 0.0F && f1 < 3.4028235E37F) {
                     serverPlayer.awardStat(Stats.DAMAGE_BLOCKED_BY_SHIELD, Math.round(f1 * 10.0F));
                 }
             }
 
             if (damageSource.getEntity() instanceof ServerPlayer serverPlayerx) {
-                CriteriaTriggers.PLAYER_HURT_ENTITY.trigger(serverPlayerx, this, damageSource, amount, amount, flag);
+                CriteriaTriggers.PLAYER_HURT_ENTITY.trigger(serverPlayerx, this, damageSource, originAmount, amount, flag);
             }
 
             return flag2;
