--- a/net/minecraft/util/datafix/fixes/EntityIdFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityIdFix.java
@@ -97,8 +97,8 @@
 
     @Override
     public TypeRewriteRule makeRule() {
-        TaggedChoiceType<String> taggedChoiceType = this.getInputSchema().findChoiceType(References.ENTITY);
-        TaggedChoiceType<String> taggedChoiceType1 = this.getOutputSchema().findChoiceType(References.ENTITY);
+        TaggedChoiceType<String> taggedChoiceType = (TaggedChoiceType<String>) this.getInputSchema().findChoiceType(References.ENTITY);
+        TaggedChoiceType<String> taggedChoiceType1 = (TaggedChoiceType<String>) this.getOutputSchema().findChoiceType(References.ENTITY);
         Type<?> type = this.getInputSchema().getType(References.ITEM_STACK);
         Type<?> type1 = this.getOutputSchema().getType(References.ITEM_STACK);
         return TypeRewriteRule.seq(
