--- a/net/minecraft/util/datafix/fixes/ItemRemoveBlockEntityTagFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemRemoveBlockEntityTagFix.java
@@ -43,7 +43,7 @@
                             return typed;
                         }
 
-                        return typed.set(opticFinder1, ((Pair)optional2.get()).getFirst());
+                        return typed.set(opticFinder1, optional2.get().getFirst());
                     }
                 }
             }
