--- a/net/minecraft/world/level/Explosion.java
+++ b/net/minecraft/world/level/Explosion.java
@@ -320,14 +320,14 @@
     }
 
     private static void addBlockDrops(ObjectArrayList<Pair<ItemStack, BlockPos>> stacks, ItemStack stack, BlockPos pos) {
-        int size = stacks.size();
+        int i = stacks.size();
 
-        for(int i = 0; i < size; ++i) {
-            Pair<ItemStack, BlockPos> pair = stacks.get(i);
+        for(int i1 = 0; i1 < i; ++i1) {
+            Pair<ItemStack, BlockPos> pair = stacks.get(i1);
             ItemStack itemStack = pair.getFirst();
             if (ItemEntity.areMergable(itemStack, stack)) {
                 ItemStack itemStack1 = ItemEntity.merge(itemStack, stack, 16);
-                stacks.set(i, Pair.of(itemStack1, pair.getSecond()));
+                stacks.set(i1, Pair.of(itemStack1, pair.getSecond()));
                 if (stack.isEmpty()) {
                     return;
                 }
