--- a/net/minecraft/world/level/GameRules.java
+++ b/net/minecraft/world/level/GameRules.java
@@ -239,8 +239,8 @@
     private static <T extends GameRules.Value<T>> void callVisitorCap(
         GameRules.GameRuleTypeVisitor gameRuleTypeVisitor, GameRules.Key<?> key, GameRules.Type<?> type
     ) {
-        gameRuleTypeVisitor.visit(key, type);
-        type.callVisitor(gameRuleTypeVisitor, key);
+        gameRuleTypeVisitor.visit((Key<T>) key, (GameRules.Type<T>) type);
+        ((GameRules.Type<T>) type).callVisitor(gameRuleTypeVisitor, (Key<T>) key);
     }
 
     public void assignFrom(GameRules rules, @Nullable MinecraftServer minecraftServer) {
