--- a/net/minecraft/util/datafix/fixes/TrappedChestBlockEntityFix.java
+++ b/net/minecraft/util/datafix/fixes/TrappedChestBlockEntityFix.java
@@ -80,7 +80,7 @@
                                         Dynamic<?> dynamic = typed1.get(DSL.remainderFinder());
                                         int _int = dynamic.get("xPos").asInt(0);
                                         int _int1 = dynamic.get("zPos").asInt(0);
-                                        TaggedChoiceType<String> taggedChoiceType = this.getInputSchema().findChoiceType(References.BLOCK_ENTITY);
+                                        TaggedChoiceType<String> taggedChoiceType = (TaggedChoiceType<String>) this.getInputSchema().findChoiceType(References.BLOCK_ENTITY);
                                         return typed1.updateTyped(
                                             opticFinder,
                                             typed3 -> typed3.updateTyped(
