--- a/net/minecraft/server/commands/ForceLoadCommand.java
+++ b/net/minecraft/server/commands/ForceLoadCommand.java
@@ -198,7 +_,7 @@
                         source.sendSuccess(
                             () -> Component.translatable(
                                 "commands.forceload." + (add ? "added" : "removed") + ".multiple",
-                                i2,
+                                i2x,
                                 Component.translationArg(resourceKey.location()),
                                 Component.translationArg(chunkPos2),
                                 Component.translationArg(chunkPos3)
