--- a/net/minecraft/world/item/BrushItem.java
+++ b/net/minecraft/world/item/BrushItem.java
@@ -67,9 +67,8 @@
                     BlockState blockState = world.getBlockState(blockPos);
                     HumanoidArm humanoidArm = user.getUsedItemHand() == InteractionHand.MAIN_HAND ? player.getMainArm() : player.getMainArm().getOpposite();
                     this.spawnDustParticles(world, blockHitResult, blockState, user.getViewVector(0.0F), humanoidArm);
-                    Block flag1 = blockState.getBlock();
                     SoundEvent brushSound;
-                    if (flag1 instanceof BrushableBlock brushableBlock) {
+                    if (blockState.getBlock() instanceof BrushableBlock brushableBlock) {
                         brushSound = brushableBlock.getBrushSound();
                     } else {
                         brushSound = SoundEvents.BRUSH_GENERIC;
@@ -77,8 +76,7 @@
 
                     world.playSound(player, blockPos, brushSound, SoundSource.BLOCKS);
                     if (!world.isClientSide()) {
-                        BlockEntity var18 = world.getBlockEntity(blockPos);
-                        if (var18 instanceof BrushableBlockEntity brushableBlockEntity) {
+                        if (world.getBlockEntity(blockPos) instanceof BrushableBlockEntity brushableBlockEntity) {
                             boolean flag1 = brushableBlockEntity.brush(world.getGameTime(), player, blockHitResult.getDirection());
                             if (flag1) {
                                 EquipmentSlot equipmentSlot = stack.equals(player.getItemBySlot(EquipmentSlot.OFFHAND))
