--- a/net/minecraft/world/entity/ai/behavior/EraseMemoryIf.java
+++ b/net/minecraft/world/entity/ai/behavior/EraseMemoryIf.java
@@ -9,7 +9,7 @@
 public class EraseMemoryIf {
     public static <E extends LivingEntity> BehaviorControl<E> create(Predicate<E> shouldEraseMemory, MemoryModuleType<?> erasingMemory) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor>group(instance.present(erasingMemory)).apply(instance, memory -> (level, entity, gameTime) -> {
+            instance -> instance.group(instance.present(erasingMemory)).apply(instance, memory -> (level, entity, gameTime) -> {
                         if (shouldEraseMemory.test(entity)) {
                             memory.erase();
                             return true;
