--- a/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
+++ b/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
@@ -13,7 +13,7 @@
 public class InsideBrownianWalk {
     public static BehaviorControl<PathfinderMob> create(float speedModifier) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor>group(instance.absent(MemoryModuleType.WALK_TARGET))
+            instance -> instance.group(instance.absent(MemoryModuleType.WALK_TARGET))
                     .apply(
                         instance,
                         walkTarget -> (level, mob, gameTime) -> {
