--- a/net/minecraft/world/entity/ai/behavior/RingBell.java
+++ b/net/minecraft/world/entity/ai/behavior/RingBell.java
@@ -16,7 +16,7 @@
 
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor>group(instance.present(MemoryModuleType.MEETING_POINT))
+            instance -> instance.group(instance.present(MemoryModuleType.MEETING_POINT))
                     .apply(instance, meetingPoint -> (level, entity, gameTime) -> {
                             if (level.random.nextFloat() <= 0.95F) {
                                 return false;
