--- a/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.java
+++ b/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.java
@@ -16,7 +16,7 @@
         MemoryModuleType<GlobalPos> blockTargetMemory, float speedModifier, int closeEnoughDist, int tooFarDistance, int tooLongUnreachableDuration
     ) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.registered(MemoryModuleType.CANT_REACH_WALK_TARGET_SINCE),
                         instance.absent(MemoryModuleType.WALK_TARGET),
                         instance.present(blockTargetMemory)
