--- a/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromAttackTargetIfTargetOutOfReach.java
+++ b/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromAttackTargetIfTargetOutOfReach.java
@@ -20,7 +20,7 @@
 
     public static BehaviorControl<Mob> create(Function<LivingEntity, Float> speedModifier) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.registered(MemoryModuleType.WALK_TARGET),
                         instance.registered(MemoryModuleType.LOOK_TARGET),
                         instance.present(MemoryModuleType.ATTACK_TARGET),
