--- a/net/minecraft/world/entity/ai/behavior/StartCelebratingIfTargetDead.java
+++ b/net/minecraft/world/entity/ai/behavior/StartCelebratingIfTargetDead.java
@@ -11,7 +11,7 @@
 public class StartCelebratingIfTargetDead {
     public static BehaviorControl<LivingEntity> create(int duration, BiPredicate<LivingEntity, LivingEntity> canDance) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.ATTACK_TARGET),
                         instance.registered(MemoryModuleType.ANGRY_AT),
                         instance.absent(MemoryModuleType.CELEBRATE_LOCATION),
