--- a/net/minecraft/world/entity/ai/behavior/StrollToPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollToPoi.java
@@ -13,7 +13,7 @@
     public static BehaviorControl<PathfinderMob> create(MemoryModuleType<GlobalPos> poiPosMemory, float speedModifier, int closeEnoughDist, int maxDistFromPoi) {
         MutableLong mutableLong = new MutableLong(0L);
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor>group(instance.registered(MemoryModuleType.WALK_TARGET), instance.present(poiPosMemory))
+            instance -> instance.group(instance.registered(MemoryModuleType.WALK_TARGET), instance.present(poiPosMemory))
                     .apply(instance, (walkTarget, poiPos) -> (level, mob, gameTime) -> {
                             GlobalPos globalPos = instance.get(poiPos);
                             if (level.dimension() != globalPos.dimension() || !globalPos.pos().closerToCenterThan(mob.position(), (double)maxDistFromPoi)) {
