--- a/net/minecraft/world/entity/ai/behavior/TryFindLandNearWater.java
+++ b/net/minecraft/world/entity/ai/behavior/TryFindLandNearWater.java
@@ -17,7 +17,7 @@
     public static BehaviorControl<PathfinderMob> create(int range, float speedModifier) {
         MutableLong mutableLong = new MutableLong(0L);
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.absent(MemoryModuleType.ATTACK_TARGET),
                         instance.absent(MemoryModuleType.WALK_TARGET),
                         instance.registered(MemoryModuleType.LOOK_TARGET)
