--- a/net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
+++ b/net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
@@ -23,7 +23,7 @@
 
     public static OneShot<PathfinderMob> create(float speedModifier, int maxHorizontalDist, int maxVericalDist) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor>group(instance.absent(MemoryModuleType.WALK_TARGET))
+            instance -> instance.group(instance.absent(MemoryModuleType.WALK_TARGET))
                     .apply(
                         instance,
                         walkTarget -> (level, mob, gameTime) -> {
