--- a/net/minecraft/world/entity/monster/warden/WardenAi.java
+++ b/net/minecraft/world/entity/monster/warden/WardenAi.java
@@ -132,7 +_,7 @@
         brain.addActivityWithConditions(
             Activity.ACTING,
             10,
-            ImmutableList.of(
+            ImmutableList.<BehaviorControl<? super Warden>>of(
                 new LookAtTargetSink(45, 90),
                 new MoveToTargetSink(500, 600),
                 new StageMinusOne<>(6000, Pose.SLEEPING),
