--- a/net/minecraft/world/entity/ai/behavior/warden/SetRoarTarget.java
+++ b/net/minecraft/world/entity/ai/behavior/warden/SetRoarTarget.java
@@ -13,7 +13,7 @@
 public class SetRoarTarget {
     public static <E extends Warden> BehaviorControl<E> create(Function<E, Optional<? extends LivingEntity>> targetFinder) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.absent(MemoryModuleType.ROAR_TARGET),
                         instance.absent(MemoryModuleType.ATTACK_TARGET),
                         instance.registered(MemoryModuleType.CANT_REACH_WALK_TARGET_SINCE)
