--- a/net/minecraft/world/entity/monster/warden/WardenAi.java
+++ b/net/minecraft/world/entity/monster/warden/WardenAi.java
@@ -83,7 +83,7 @@
         MemoryModuleType.SONIC_BOOM_SOUND_DELAY
     );
     private static final BehaviorControl<Warden> DIG_COOLDOWN_SETTER = BehaviorBuilder.create(
-        context -> context.<MemoryAccessor>group(context.registered(MemoryModuleType.DIG_COOLDOWN)).apply(context, digCooldown -> (world, entity, time) -> {
+        context -> context.group(context.registered(MemoryModuleType.DIG_COOLDOWN)).apply(context, digCooldown -> (world, entity, time) -> {
                     if (context.tryGet(digCooldown).isPresent()) {
                         digCooldown.setWithExpiry(Unit.INSTANCE, 1200L);
                     }
