--- a/net/minecraft/util/datafix/DataFixTypes.java
+++ b/net/minecraft/util/datafix/DataFixTypes.java
@@ -43,7 +43,7 @@
     }
 
     public CompoundTag update(DataFixer dataFixer, CompoundTag nbt, int oldVersion, int i) {
-        return this.update(dataFixer, new Dynamic<>(NbtOps.INSTANCE, nbt), oldVersion, i).getValue();
+        return (CompoundTag) this.update(dataFixer, new Dynamic<>(NbtOps.INSTANCE, nbt), oldVersion, i).getValue();
     }
 
     public CompoundTag updateToCurrentVersion(DataFixer dataFixer, CompoundTag nbt, int i) {
