--- a/net/minecraft/commands/arguments/EntityArgument.java
+++ b/net/minecraft/commands/arguments/EntityArgument.java
@@ -117,9 +117,9 @@
 
     @Override
     public <S> CompletableFuture<Suggestions> listSuggestions(CommandContext<S> context, SuggestionsBuilder builder) {
-        StringReader stringreader = context.getSource();
-        if (stringreader instanceof SharedSuggestionProvider sharedsuggestionprovider) {
-            stringreader = new StringReader(builder.getInput());
+        S s = context.getSource();
+        if (s instanceof SharedSuggestionProvider sharedsuggestionprovider) {
+            StringReader stringreader = new StringReader(builder.getInput());
             stringreader.setCursor(builder.getStart());
             EntitySelectorParser entityselectorparser = new EntitySelectorParser(stringreader, sharedsuggestionprovider.hasPermission(2));
 
