--- a/net/minecraft/world/entity/animal/axolotl/ValidatePlayDead.java
+++ b/net/minecraft/world/entity/animal/axolotl/ValidatePlayDead.java
@@ -9,7 +9,7 @@
 public class ValidatePlayDead {
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.PLAY_DEAD_TICKS), instance.registered(MemoryModuleType.HURT_BY_ENTITY)
                     )
                     .apply(instance, (playDeadTicks, hurtByEntity) -> (world, entity, time) -> {
