--- a/net/minecraft/world/entity/SpawnPlacements.java
+++ b/net/minecraft/world/entity/SpawnPlacements.java
@@ -73,7 +73,7 @@
         EntityType<T> entityType, ServerLevelAccessor serverLevel, MobSpawnType spawnType, BlockPos pos, RandomSource random
     ) {
         SpawnPlacements.Data data = DATA_BY_TYPE.get(entityType);
-        return data == null || data.predicate.test(entityType, serverLevel, spawnType, pos, random);
+        return data == null || data.predicate.test((EntityType) entityType, serverLevel, spawnType, pos, random);
     }
 
     static {
