--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -1170,6 +_,7 @@
             }
 
             ItemStack itemInUse = this.getUseItem();
+            float originalDamage = damage;
             float damageBlocked = this.applyItemBlocking(level, source, damage);
             damage -= damageBlocked;
             boolean blocked = damageBlocked > 0.0F;
@@ -1261,14 +_,14 @@
             }
 
             if (this instanceof ServerPlayer serverPlayer) {
-                CriteriaTriggers.ENTITY_HURT_PLAYER.trigger(serverPlayer, source, damage, damage, blocked);
+                CriteriaTriggers.ENTITY_HURT_PLAYER.trigger(serverPlayer, source, originalDamage, damage, blocked);
                 if (damageBlocked > 0.0F && damageBlocked < 3.4028235E37F) {
                     serverPlayer.awardStat(Stats.DAMAGE_BLOCKED_BY_SHIELD, Math.round(damageBlocked * 10.0F));
                 }
             }
 
             if (source.getEntity() instanceof ServerPlayer sourcePlayer) {
-                CriteriaTriggers.PLAYER_HURT_ENTITY.trigger(sourcePlayer, this, source, damage, damage, blocked);
+                CriteriaTriggers.PLAYER_HURT_ENTITY.trigger(sourcePlayer, this, source, originalDamage, damage, blocked);
             }
 
             return success;
