--- a/net/minecraft/world/entity/animal/goat/GoatAi.java
+++ b/net/minecraft/world/entity/animal/goat/GoatAi.java
@@ -82,7 +_,7 @@
             0,
             ImmutableList.of(
                 new Swim<>(0.8F),
-                new AnimalPanic(2.0F),
+                new AnimalPanic<>(2.0F),
                 new LookAtTargetSink(45, 90),
                 new MoveToTargetSink(),
                 new CountDownCooldownTicks(MemoryModuleType.TEMPTATION_COOLDOWN_TICKS),
