--- a/net/minecraft/world/level/storage/loot/functions/EnchantRandomlyFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/EnchantRandomlyFunction.java
@@ -53,7 +53,7 @@
     public ItemStack run(ItemStack stack, LootContext context) {
         RandomSource random = context.getRandom();
         Optional<Holder<Enchantment>> optional = this.enchantments
-            .<Holder.Reference<Enchantment>>flatMap(holderSet -> holderSet.getRandomElement(random))
+            .flatMap(holderSet -> holderSet.getRandomElement(random))
             .or(
                 () -> {
                     boolean isBook = stack.is(Items.BOOK);
