--- a/net/minecraft/network/syncher/EntityDataSerializer.java
+++ b/net/minecraft/network/syncher/EntityDataSerializer.java
@@ -13,7 +_,7 @@
     T copy(T value);
 
     static <T> EntityDataSerializer<T> forValueType(StreamCodec<? super RegistryFriendlyByteBuf, T> streamCodec) {
-        return () -> streamCodec;
+        return (ForValueType<T>) () -> streamCodec;
     }
 
     public interface ForValueType<T> extends EntityDataSerializer<T> {
