--- a/net/minecraft/world/entity/monster/hoglin/HoglinAi.java
+++ b/net/minecraft/world/entity/monster/hoglin/HoglinAi.java
@@ -112,7 +_,7 @@
                 SetWalkTargetAwayFrom.entity(MemoryModuleType.AVOID_TARGET, 1.3F, 15, false),
                 createIdleMovementBehaviors(),
                 SetEntityLookTargetSometimes.create(8.0F, UniformInt.of(30, 60)),
-                EraseMemoryIf.<PathfinderMob>create(HoglinAi::wantsToStopFleeing, MemoryModuleType.AVOID_TARGET)
+                EraseMemoryIf.create(HoglinAi::wantsToStopFleeing, MemoryModuleType.AVOID_TARGET)
             ),
             MemoryModuleType.AVOID_TARGET
         );
