--- a/net/minecraft/network/protocol/configuration/ClientboundUpdateEnabledFeaturesPacket.java
+++ b/net/minecraft/network/protocol/configuration/ClientboundUpdateEnabledFeaturesPacket.java
@@ -8,7 +8,7 @@
 
 public record ClientboundUpdateEnabledFeaturesPacket(Set<ResourceLocation> features) implements Packet<ClientConfigurationPacketListener> {
     public ClientboundUpdateEnabledFeaturesPacket(FriendlyByteBuf buffer) {
-        this(buffer.readCollection(HashSet::new, FriendlyByteBuf::readResourceLocation));
+        this(buffer.<ResourceLocation, Set<ResourceLocation>>readCollection(HashSet::new, FriendlyByteBuf::readResourceLocation));
     }
 
     @Override
