--- a/net/minecraft/world/level/GameRules.java
+++ b/net/minecraft/world/level/GameRules.java
@@ -235,8 +235,8 @@
     }
 
     private static <T extends GameRules.Value<T>> void callVisitorCap(GameRules.GameRuleTypeVisitor consumer, GameRules.Key<?> key, GameRules.Type<?> type) {
-        consumer.visit(key, type);
-        type.callVisitor(consumer, key);
+        consumer.visit((Key<T>) key, (Type<T>) type);
+        ((Type<T>) type).callVisitor(consumer, (Key<T>) key);
     }
 
     public void assignFrom(GameRules rules, @Nullable MinecraftServer server) {
