--- a/net/minecraft/server/commands/PlaceCommand.java
+++ b/net/minecraft/server/commands/PlaceCommand.java
@@ -71,14 +_,14 @@
                 .then(
                     Commands.literal("feature")
                         .then(
-                            ((RequiredArgumentBuilder)Commands.argument("feature", ResourceOrIdArgument.feature(context))
+                            Commands.argument("feature", ResourceOrIdArgument.feature(context))
                                     .executes(
                                         c -> placeFeature(
                                             (CommandSourceStack)c.getSource(),
                                             ResourceOrIdArgument.getFeature(c, "feature"),
                                             BlockPos.containing(((CommandSourceStack)c.getSource()).getPosition())
                                         )
-                                    ))
+                                    )
                                 .then(
                                     Commands.argument("pos", BlockPosArgument.blockPos())
                                         .executes(
