--- a/net/minecraft/data/recipes/RecipeProvider.java
+++ b/net/minecraft/data/recipes/RecipeProvider.java
@@ -659,7 +659,7 @@
     }
 
     private static InventoryChangeTrigger.TriggerInstance inventoryTrigger(ItemPredicate.Builder... builders) {
-        return inventoryTrigger(Arrays.stream(builders).flatMap(builder -> builder.build().stream()).toArray(i -> new ItemPredicate[i]));
+        return inventoryTrigger(Arrays.stream(builders).flatMap(builder -> builder.build().stream()).toArray(ItemPredicate[]::new));
     }
 
     private static InventoryChangeTrigger.TriggerInstance inventoryTrigger(ItemPredicate... predicates) {
