--- a/net/minecraft/core/BlockPos.java
+++ b/net/minecraft/core/BlockPos.java
@@ -389,7 +389,7 @@
         return StreamSupport.stream(betweenClosed(startX, startY, startZ, endX, endY, endZ).spliterator(), false);
     }
 
-    public static Iterable<BlockPos> betweenClosed(int startX, int startY, int startZ, int endX, int endY, int endZ) {
+    public static Iterable<BlockPos> b(int startX, int startY, int startZ, int endX, int endY, int endZ) {
         int i = endX - startX + 1;
         int i1 = endY - startY + 1;
         int i2 = endZ - startZ + 1;
