--- a/net/minecraft/core/Registry.java
+++ b/net/minecraft/core/Registry.java
@@ -155,7 +155,7 @@
     Optional<HolderSet.Named<T>> getTag(TagKey<T> tag);
 
     default Iterable<Holder<T>> getTagOrEmpty(TagKey<T> tag) {
-        return DataFixUtils.orElse(this.getTag(tag), List.<T>of());
+        return DataFixUtils.orElse(this.getTag(tag), List.of());
     }
 
     HolderSet.Named<T> getOrCreateTag(TagKey<T> tag);
@@ -178,7 +178,7 @@
             @Nullable
             @Override
             public Holder<T> byId(int i) {
-                return (Holder<T>)Registry.this.getHolder(i).orElse((T)null);
+                return (Holder<T>)Registry.this.getHolder(i).orElse(null);
             }
 
             @Override
@@ -188,7 +188,7 @@
 
             @Override
             public Iterator<Holder<T>> iterator() {
-                return Registry.this.holders().map(entry -> entry).iterator();
+                return Registry.this.holders().map(entry -> (Holder<T>) entry).iterator();
             }
         };
     }
