--- a/net/minecraft/core/RegistrySetBuilder.java
+++ b/net/minecraft/core/RegistrySetBuilder.java
@@ -268,8 +268,8 @@
                 ResourceKey<?> resourceKey = entry.getKey();
                 if (resourceKey.isFor(this.key)) {
                     RegistrySetBuilder.RegisteredValue<T> registeredValue = (RegistrySetBuilder.RegisteredValue)entry.getValue();
-                    Holder.Reference<T> reference = buildState.lookup.holders.remove(resourceKey);
-                    map.put(resourceKey, new RegistrySetBuilder.ValueAndHolder<>(registeredValue, Optional.ofNullable(reference)));
+                    Holder.Reference<T> reference = (Holder.Reference<T>) buildState.lookup.holders.remove(resourceKey);
+                    map.put((ResourceKey<T>) resourceKey, new RegistrySetBuilder.ValueAndHolder<>(registeredValue, Optional.ofNullable(reference)));
                     iterator.remove();
                 }
             }
@@ -291,7 +291,7 @@
         }
 
         <T> Holder.Reference<T> getOrCreate(ResourceKey<T> resourceKey) {
-            return this.holders.computeIfAbsent(resourceKey, resourceKey1 -> Holder.Reference.createStandAlone(this.owner, resourceKey1));
+            return ((Map<ResourceKey<T>, Holder.Reference<T>>) (Map) this.holders).computeIfAbsent(resourceKey, resourceKey1 -> Holder.Reference.<T>createStandAlone((HolderOwner<T>) this.owner, resourceKey1));
         }
     }
 
