--- a/net/minecraft/TracingExecutor.java
+++ b/net/minecraft/TracingExecutor.java
@@ -21,7 +_,7 @@
                 }
             });
         } else {
-            return (Executor)(TracyClient.isAvailable() ? runnable -> this.service.execute(() -> {
+            return (TracyClient.isAvailable() ? runnable -> this.service.execute(() -> {
                 try (Zone zone = TracyClient.beginZone(string, SharedConstants.IS_RUNNING_IN_IDE)) {
                     runnable.run();
                 }
