--- a/net/minecraft/world/level/block/Block.java
+++ b/net/minecraft/world/level/block/Block.java
@@ -558,7 +_,7 @@
     }
 
     private static <S extends StateHolder<?, S>, T extends Comparable<T>> S setValueHelper(S stateHolder, Property<T> property, Object value) {
-        return stateHolder.setValue(property, (Comparable)value);
+        return stateHolder.setValue(property, (T) value);
     }
 
     @Deprecated
