--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -587,7 +587,7 @@
                 }
 
                 return isChunkDataValid;
-            })).thenApplyAsync(optional -> {
+            })).<Either<ChunkAccess, ChunkHolder.ChunkLoadingFailure>>thenApplyAsync(optional -> {
             this.level.getProfiler().incrementCounter("chunkLoad");
             if (optional.isPresent()) {
                 ChunkAccess chunkAccess = ChunkSerializer.read(this.level, this.poiManager, chunkPos, optional.get());
