--- a/net/minecraft/nbt/NbtUtils.java
+++ b/net/minecraft/nbt/NbtUtils.java
@@ -350,7 +350,7 @@
             case 9:
                 ListTag listTag = (ListTag)nbt;
                 int size = listTag.size();
-                int i = listTag.getElementType();
+                i = listTag.getElementType();
                 String string = i == 0 ? "undefined" : TagTypes.getType(i).getPrettyName();
                 indent(depth, stringBuilder).append("list<").append(string).append(">[").append(size).append("] [");
                 if (size != 0) {
@@ -382,8 +382,8 @@
                     indent(depth + 1, stringBuilder);
                 }
 
-                int i = list.stream().mapToInt(String::length).max().orElse(0);
-                String string = Strings.repeat(" ", i);
+                i = list.stream().mapToInt(String::length).max().orElse(0);
+                string = Strings.repeat(" ", i);
 
                 for(int i2 = 0; i2 < list.size(); ++i2) {
                     if (i2 != 0) {
@@ -404,7 +404,7 @@
             case 11:
                 IntArrayTag intArrayTag = (IntArrayTag)nbt;
                 int[] asIntArray = intArrayTag.getAsIntArray();
-                int i = 0;
+                i = 0;
 
                 for(int i3 : asIntArray) {
                     i = Math.max(i, String.format(Locale.ROOT, "%X", i3).length());
