--- a/net/minecraft/world/entity/ai/behavior/warden/TryToSniff.java
+++ b/net/minecraft/world/entity/ai/behavior/warden/TryToSniff.java
@@ -16,7 +16,7 @@
 
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.registered(MemoryModuleType.IS_SNIFFING),
                         instance.registered(MemoryModuleType.WALK_TARGET),
                         instance.absent(MemoryModuleType.SNIFF_COOLDOWN),
