--- a/net/minecraft/world/item/BrushItem.java
+++ b/net/minecraft/world/item/BrushItem.java
@@ -74,9 +74,8 @@
                         this.spawnDustParticles(level, blockHitResult, blockState, livingEntity.getViewVector(0.0F), humanoidArm);
                     }
 
-                    Block flag1 = blockState.getBlock();
                     SoundEvent brushSound;
-                    if (flag1 instanceof BrushableBlock brushableBlock) {
+                    if (blockState.getBlock() instanceof BrushableBlock brushableBlock) {
                         brushSound = brushableBlock.getBrushSound();
                     } else {
                         brushSound = SoundEvents.BRUSH_GENERIC;
