--- a/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
@@ -35,14 +35,14 @@
     @Override
     public void write(FriendlyByteBuf buf) {
         buf.writeVarInt(this.entity);
-        int size = this.slots.size();
+        int i = this.slots.size();
 
-        for(int i = 0; i < size; ++i) {
-            Pair<EquipmentSlot, ItemStack> pair = this.slots.get(i);
+        for(int i1 = 0; i1 < i; ++i1) {
+            Pair<EquipmentSlot, ItemStack> pair = this.slots.get(i1);
             EquipmentSlot equipmentSlot = pair.getFirst();
-            boolean flag = i != size - 1;
-            int i1 = equipmentSlot.ordinal();
-            buf.writeByte(flag ? i1 | -128 : i1);
+            boolean flag = i1 != i - 1;
+            int i2 = equipmentSlot.ordinal();
+            buf.writeByte(flag ? i2 | -128 : i2);
             buf.writeItem(pair.getSecond());
         }
     }
