--- a/net/minecraft/network/protocol/game/ClientboundUpdateRecipesPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundUpdateRecipesPacket.java
@@ -46,6 +46,6 @@
     public static <T extends Recipe<?>> void toNetwork(FriendlyByteBuf buf, T recipe) {
         buf.writeResourceLocation(BuiltInRegistries.RECIPE_SERIALIZER.getKey(recipe.getSerializer()));
         buf.writeResourceLocation(recipe.getId());
-        recipe.getSerializer().toNetwork(buf, recipe);
+        ((net.minecraft.world.item.crafting.RecipeSerializer<T>) recipe.getSerializer()).toNetwork(buf, recipe);
     }
 }
