--- a/net/minecraft/network/protocol/game/ClientboundCommandsPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundCommandsPacket.java
@@ -84,8 +84,8 @@
         CommandNode<SharedSuggestionProvider> commandNode;
         while((commandNode = queue.poll()) != null) {
             if (!object2IntMap.containsKey(commandNode)) {
-                int size = object2IntMap.size();
-                object2IntMap.put(commandNode, size);
+                int i = object2IntMap.size();
+                object2IntMap.put(commandNode, i);
                 queue.addAll(commandNode.getChildren());
                 if (commandNode.getRedirect() != null) {
                     queue.add(commandNode.getRedirect());
