--- a/net/minecraft/world/entity/EntityReference.java
+++ b/net/minecraft/world/entity/EntityReference.java
@@ -93,7 +_,7 @@
 
     @Nullable
     public static <StoredEntityType extends UniquelyIdentifyable> EntityReference<StoredEntityType> read(ValueInput valueInput, String string) {
-        return valueInput.read(string, codec()).orElse(null);
+        return valueInput.read(string, EntityReference.<StoredEntityType>codec()).orElse(null);
     }
 
     @Nullable
@@ -105,7 +_,7 @@
             ? new EntityReference<>(optional.get())
             : valueInput.getString(string)
                 .map(string1 -> OldUsersConverter.convertMobOwnerIfNecessary(level.getServer(), string1))
-                .map(EntityReference::new)
+                .map(EntityReference<StoredEntityType>::new)
                 .orElse(null);
     }
 
