--- a/net/minecraft/world/item/alchemy/Potion.java
+++ b/net/minecraft/world/item/alchemy/Potion.java
@@ -16,7 +_,7 @@
 
     public static Holder<Potion> byName(String string) {
         ResourceLocation resourceLocation = ResourceLocation.tryParse(string);
-        return resourceLocation == null ? Potions.EMPTY : BuiltInRegistries.POTION.getHolder(resourceLocation).map(Function.identity()).orElse(Potions.EMPTY);
+        return resourceLocation == null ? Potions.EMPTY : BuiltInRegistries.POTION.getHolder(resourceLocation).map(Function.<Holder<Potion>>identity()).orElse(Potions.EMPTY);
     }
 
     public Potion(MobEffectInstance... effects) {
