--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -622,8 +622,7 @@
         } else if (candidate.getItem() instanceof CrossbowItem && existing.getItem() instanceof CrossbowItem) {
             return this.canReplaceEqualItem(candidate, existing);
         } else {
-            Item diggerItem = candidate.getItem();
-            if (diggerItem instanceof ArmorItem armorItem) {
+            if (candidate.getItem() instanceof ArmorItem armorItem) {
                 if (EnchantmentHelper.hasBindingCurse(existing)) {
                     return false;
                 } else if (!(existing.getItem() instanceof ArmorItem)) {
@@ -644,8 +643,7 @@
                         return true;
                     }
 
-                    Item diggerItem1 = existing.getItem();
-                    if (diggerItem1 instanceof DiggerItem diggerItem) {
+                    if (existing.getItem() instanceof DiggerItem diggerItem) {
                         DiggerItem diggerItem1x = (DiggerItem)candidate.getItem();
                         if (diggerItem1x.getAttackDamage() != diggerItem.getAttackDamage()) {
                             return diggerItem1x.getAttackDamage() > diggerItem.getAttackDamage();
@@ -1125,7 +1123,7 @@
                 if (this.level() instanceof ServerLevel) {
                     SpawnEggItem spawnEggItem = (SpawnEggItem)itemInHand.getItem();
                     Optional<Mob> optional = spawnEggItem.spawnOffspringFromSpawnEgg(
-                        player, this, this.getType(), (ServerLevel)this.level(), this.position(), itemInHand
+                        player, this, (EntityType<Mob>) this.getType(), (ServerLevel)this.level(), this.position(), itemInHand
                     );
                     optional.ifPresent(mob -> this.onOffspringSpawnedFromEgg(player, mob));
                     return optional.isPresent() ? InteractionResult.SUCCESS : InteractionResult.PASS;
