--- a/net/minecraft/commands/synchronization/SuggestionProviders.java
+++ b/net/minecraft/commands/synchronization/SuggestionProviders.java
@@ -48,7 +_,7 @@
             throw new IllegalArgumentException("A command suggestion provider is already registered with the name " + name);
         } else {
             PROVIDERS_BY_NAME.put(name, provider);
-            return new SuggestionProviders.Wrapper(name, provider);
+            return (SuggestionProvider<S>) new Wrapper(name, provider);
         }
     }
 
