--- a/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
+++ b/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
@@ -45,7 +45,7 @@
         MutableLong mutablelong = new MutableLong(0L);
         Long2ObjectMap<AcquirePoi.JitteredLinearRetry> long2objectmap = new Long2ObjectOpenHashMap<>();
         OneShot<PathfinderMob> oneshot = BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor>group(instance.absent(potentialPoiPosModule))
+            instance -> instance.group(instance.absent(potentialPoiPosModule))
                     .apply(
                         instance,
                         queryResult -> (world, entity, time) -> {
@@ -101,7 +101,7 @@
         );
         return potentialPoiPosModule == poiPosModule
             ? oneshot
-            : BehaviorBuilder.create(context -> context.<MemoryAccessor>group(context.absent(poiPosModule)).apply(context, poiPos -> oneshot));
+            : BehaviorBuilder.create(context -> context.group(context.absent(poiPosModule)).apply(context, poiPos -> oneshot));
     }
 
     @Nullable
