--- a/net/minecraft/world/entity/ai/behavior/CopyMemoryWithExpiry.java
+++ b/net/minecraft/world/entity/ai/behavior/CopyMemoryWithExpiry.java
@@ -12,7 +12,7 @@
         Predicate<E> canCopyMemory, MemoryModuleType<? extends T> sourceMemory, MemoryModuleType<T> targetMemory, UniformInt durationOfCopy
     ) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor>group(instance.present(sourceMemory), instance.absent(targetMemory))
+            instance -> instance.group(instance.present(sourceMemory), instance.absent(targetMemory))
                     .apply(instance, (source, target) -> (level, entity, gameTime) -> {
                             if (!canCopyMemory.test(entity)) {
                                 return false;
