--- a/net/minecraft/world/entity/ai/behavior/GoToTargetLocation.java
+++ b/net/minecraft/world/entity/ai/behavior/GoToTargetLocation.java
@@ -19,7 +19,7 @@
 
     public static <E extends Mob> OneShot<E> create(MemoryModuleType<BlockPos> locationMemory, int closeEnoughDist, float speedModifier) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(locationMemory),
                         instance.absent(MemoryModuleType.ATTACK_TARGET),
                         instance.absent(MemoryModuleType.WALK_TARGET),
