--- a/net/minecraft/world/entity/ai/behavior/MeleeAttack.java
+++ b/net/minecraft/world/entity/ai/behavior/MeleeAttack.java
@@ -13,7 +13,7 @@
 public class MeleeAttack {
     public static OneShot<Mob> create(int cooldownBetweenAttacks) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.registered(MemoryModuleType.LOOK_TARGET),
                         instance.present(MemoryModuleType.ATTACK_TARGET),
                         instance.absent(MemoryModuleType.ATTACK_COOLING_DOWN),
