--- a/net/minecraft/world/entity/ai/behavior/MoveToSkySeeingSpot.java
+++ b/net/minecraft/world/entity/ai/behavior/MoveToSkySeeingSpot.java
@@ -16,7 +16,7 @@
 public class MoveToSkySeeingSpot {
     public static OneShot<LivingEntity> create(float speed) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor>group(context.absent(MemoryModuleType.WALK_TARGET)).apply(context, walkTarget -> (world, entity, time) -> {
+            context -> context.group(context.absent(MemoryModuleType.WALK_TARGET)).apply(context, walkTarget -> (world, entity, time) -> {
                         if (world.canSeeSky(entity.blockPosition())) {
                             return false;
                         } else {
