--- a/net/minecraft/world/entity/ai/behavior/SetHiddenState.java
+++ b/net/minecraft/world/entity/ai/behavior/SetHiddenState.java
@@ -15,7 +15,7 @@
         int i = maxHiddenSeconds * 20;
         MutableInt mutableInt = new MutableInt(0);
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.present(MemoryModuleType.HIDING_PLACE), context.present(MemoryModuleType.HEARD_BELL_TIME)
                     )
                     .apply(context, (hidingPlace, heardBellTime) -> (world, entity, time) -> {
