--- a/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.java
+++ b/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.java
@@ -14,7 +14,7 @@
 public class SetWalkTargetFromBlockMemory {
     public static OneShot<Villager> create(MemoryModuleType<GlobalPos> destination, float speed, int completionRange, int maxDistance, int maxRunTime) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.registered(MemoryModuleType.CANT_REACH_WALK_TARGET_SINCE),
                         context.absent(MemoryModuleType.WALK_TARGET),
                         context.present(destination)
