--- a/net/minecraft/world/entity/ai/behavior/SocializeAtBell.java
+++ b/net/minecraft/world/entity/ai/behavior/SocializeAtBell.java
@@ -14,7 +14,7 @@
 
     public static OneShot<LivingEntity> create() {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.registered(MemoryModuleType.WALK_TARGET),
                         context.registered(MemoryModuleType.LOOK_TARGET),
                         context.present(MemoryModuleType.MEETING_POINT),
