--- a/net/minecraft/world/entity/ai/behavior/StrollToPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollToPoi.java
@@ -12,7 +12,7 @@
     public static BehaviorControl<PathfinderMob> create(MemoryModuleType<GlobalPos> posModule, float walkSpeed, int completionRange, int maxDistance) {
         MutableLong mutablelong = new MutableLong(0L);
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor>group(context.registered(MemoryModuleType.WALK_TARGET), context.present(posModule))
+            context -> context.group(context.registered(MemoryModuleType.WALK_TARGET), context.present(posModule))
                     .apply(context, (walkTarget, pos) -> (world, entity, time) -> {
                             GlobalPos globalpos = context.get(pos);
                             if (world.dimension() != globalpos.dimension() || !globalpos.pos().closerToCenterThan(entity.position(), (double)maxDistance)) {
