--- a/net/minecraft/world/entity/ai/behavior/StrollToPoiList.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollToPoiList.java
@@ -19,7 +19,7 @@
     ) {
         MutableLong mutablelong = new MutableLong(0L);
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.registered(MemoryModuleType.WALK_TARGET), context.present(secondaryPositions), context.present(primaryPosition)
                     )
                     .apply(
