--- a/net/minecraft/world/entity/ai/behavior/TryFindLandNearWater.java
+++ b/net/minecraft/world/entity/ai/behavior/TryFindLandNearWater.java
@@ -16,7 +16,7 @@
     public static BehaviorControl<PathfinderMob> create(int range, float speed) {
         MutableLong mutablelong = new MutableLong(0L);
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.absent(MemoryModuleType.ATTACK_TARGET),
                         context.absent(MemoryModuleType.WALK_TARGET),
                         context.registered(MemoryModuleType.LOOK_TARGET)
