--- a/net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.java
@@ -20,7 +20,7 @@
 
     public static BehaviorControl<LivingEntity> create(Predicate<Holder<PoiType>> poiValidator, MemoryModuleType<GlobalPos> poiPosMemory) {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor>group(instance.present(poiPosMemory)).apply(instance, poiPos -> (level, entity, gameTime) -> {
+            instance -> instance.group(instance.present(poiPosMemory)).apply(instance, poiPos -> (level, entity, gameTime) -> {
                         GlobalPos globalPos = instance.get(poiPos);
                         BlockPos blockPos = globalPos.pos();
                         if (level.dimension() == globalPos.dimension() && blockPos.closerToCenterThan(entity.position(), 16.0)) {
