--- a/net/minecraft/world/entity/ai/behavior/VillagerCalmDown.java
+++ b/net/minecraft/world/entity/ai/behavior/VillagerCalmDown.java
@@ -10,7 +10,7 @@
 
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.registered(MemoryModuleType.HURT_BY),
                         context.registered(MemoryModuleType.HURT_BY_ENTITY),
                         context.registered(MemoryModuleType.NEAREST_HOSTILE)
