--- a/net/minecraft/util/datafix/fixes/ChunkHeightAndBiomeFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkHeightAndBiomeFix.java
@@ -220,7 +_,8 @@
         } else if (ints != null && ints.length == 1024) {
             for (int i = 0; i < 16; i++) {
                 int i1 = i - lowestY;
-                dynamics[i1] = makeBiomeContainer(data, i3 -> getOldBiome(ints, i * 64 + i3));
+                final int f_i = i1;
+                dynamics[i1] = makeBiomeContainer(data, i3 -> getOldBiome(ints, f_i * 64 + i3));
             }
 
             if (overworld) {
