--- a/net/minecraft/world/level/GameRules.java
+++ b/net/minecraft/world/level/GameRules.java
@@ -256,8 +_,8 @@
     }
 
     private static <T extends GameRules.Value<T>> void callVisitorCap(GameRules.GameRuleTypeVisitor visitor, GameRules.Key<?> key, GameRules.Type<?> type) {
-        visitor.visit(key, type);
-        type.callVisitor(visitor, key);
+        visitor.visit((GameRules.Key<T>) key, (GameRules.Type<T>) type);
+        ((GameRules.Type<T>) type).callVisitor(visitor, (GameRules.Key<T>) key);
     }
 
     public void assignFrom(GameRules rules, @Nullable MinecraftServer server) {
