--- a/net/minecraft/commands/arguments/OperationArgument.java
+++ b/net/minecraft/commands/arguments/OperationArgument.java
@@ -57,7 +_,7 @@
     }
 
     private static OperationArgument.Operation getOperation(String name) throws CommandSyntaxException {
-        return (OperationArgument.Operation)(name.equals("><") ? (targetScore, sourceScore) -> {
+        return (name.equals("><") ? (targetScore, sourceScore) -> {
             int i = targetScore.get();
             targetScore.set(sourceScore.get());
             sourceScore.set(i);
