--- a/net/minecraft/server/level/ChunkTaskPriorityQueueSorter.java
+++ b/net/minecraft/server/level/ChunkTaskPriorityQueueSorter.java
@@ -154,7 +154,7 @@
         if (chunkTaskPriorityQueue == null) {
             throw (IllegalArgumentException)Util.pauseInIde(new IllegalArgumentException("No queue for: " + processor));
         } else {
-            return chunkTaskPriorityQueue;
+            return (ChunkTaskPriorityQueue<Function<ProcessorHandle<Unit>, T>>) chunkTaskPriorityQueue;
         }
     }
 
