--- a/net/minecraft/advancements/critereon/InventoryChangeTrigger.java
+++ b/net/minecraft/advancements/critereon/InventoryChangeTrigger.java
@@ -68,7 +68,7 @@
         }
 
         public static Criterion<InventoryChangeTrigger.TriggerInstance> hasItems(ItemPredicate.Builder... builders) {
-            return hasItems(Stream.of(builders).map(ItemPredicate.Builder::build).toArray(i -> new ItemPredicate[i]));
+            return hasItems(Stream.of(builders).map(ItemPredicate.Builder::build).toArray(ItemPredicate[]::new));
         }
 
         public static Criterion<InventoryChangeTrigger.TriggerInstance> hasItems(ItemPredicate... itemPredicates) {
