--- a/net/minecraft/data/tags/TagsProvider.java
+++ b/net/minecraft/data/tags/TagsProvider.java
@@ -62,6 +62,8 @@
 
     @Override
     public CompletableFuture<?> run(CachedOutput cachedOutput) {
+        record CombinedData<T>(HolderLookup.Provider contents, TagsProvider.TagLookup<T> parent) {
+        }
         return this.createContentsProvider()
             .thenApply(provider -> {
                 this.contentsDone.complete(null);
@@ -107,10 +109,6 @@
                     );
                 }
             );
-
-        record CombinedData<T>(HolderLookup.Provider contents, TagsProvider.TagLookup<T> parent) {
-        }
-
     }
 
     protected TagsProvider.TagAppender<T> tag(TagKey<T> tagKey) {
@@ -178,7 +176,7 @@
         }
 
         default boolean contains(TagKey<T> tagKey) {
-            return this.apply((T)tagKey).isPresent();
+            return this.apply(tagKey).isPresent();
         }
     }
 }
