--- a/net/minecraft/util/datafix/fixes/AddNewChoices.java
+++ b/net/minecraft/util/datafix/fixes/AddNewChoices.java
@@ -30,7 +_,7 @@
         if (type.getKeyType() != newType.getKeyType()) {
             throw new IllegalStateException("Could not inject: key type is not the same");
         } else {
-            return this.fixTypeEverywhere(name, type, (Type<Pair<K, ?>>)newType, dynamicOps -> pair -> {
+            return this.fixTypeEverywhere(name, type, (TaggedChoiceType<K>)newType, dynamicOps -> pair -> {
                     if (!((TaggedChoiceType<K>)newType).hasType(pair.getFirst())) {
                         throw new IllegalArgumentException(String.format(Locale.ROOT, "Unknown type %s in %s ", pair.getFirst(), this.type));
                     } else {
