--- a/net/minecraft/util/datafix/fixes/BlockEntityUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockEntityUUIDFix.java
@@ -23,7 +23,7 @@
             .map(dynamic -> replaceUUIDString((Dynamic<?>)dynamic, "Id", "Id").orElse((Dynamic<?>)dynamic))
             .map(dynamic -> skullTag.remove("Owner").set("SkullOwner", (Dynamic<?>)dynamic))
             .result()
-            .orElse(skullTag);
+            .orElse((Dynamic) skullTag);
     }
 
     private Dynamic<?> updateConduit(Dynamic<?> conduitTag) {
