--- a/net/minecraft/network/protocol/common/custom/GameTestClearMarkersDebugPayload.java
+++ b/net/minecraft/network/protocol/common/custom/GameTestClearMarkersDebugPayload.java
@@ -1,10 +1,9 @@
 package net.minecraft.network.protocol.common.custom;
 
-import java.lang.runtime.ObjectMethods;
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.resources.ResourceLocation;
 
-public final class GameTestClearMarkersDebugPayload extends Record implements CustomPacketPayload {
+public record GameTestClearMarkersDebugPayload() implements CustomPacketPayload {
     public static final ResourceLocation ID = new ResourceLocation("debug/game_test_clear");
 
     public GameTestClearMarkersDebugPayload(FriendlyByteBuf friendlyByteBuf) {
@@ -22,19 +21,4 @@
     public ResourceLocation id() {
         return ID;
     }
-
-    @Override
-    public final String toString() {
-        return ObjectMethods.bootstrap<"toString",GameTestClearMarkersDebugPayload,"">(this);
-    }
-
-    @Override
-    public final int hashCode() {
-        return ObjectMethods.bootstrap<"hashCode",GameTestClearMarkersDebugPayload,"">(this);
-    }
-
-    @Override
-    public final boolean equals(Object object) {
-        return ObjectMethods.bootstrap<"equals",GameTestClearMarkersDebugPayload,"">(this, object);
-    }
 }
