--- a/net/minecraft/core/HolderLookup.java
+++ b/net/minecraft/core/HolderLookup.java
@@ -87,7 +87,7 @@
             final Map<ResourceKey<? extends Registry<?>>, HolderLookup.RegistryLookup<?>> map = wrappers.collect(
                 Collectors.toUnmodifiableMap(
                     HolderLookup.RegistryLookup::key,
-                    (Function<? super HolderLookup.RegistryLookup, ? extends HolderLookup.RegistryLookup>)(wrapper -> wrapper)
+                    (wrapper -> wrapper)
                 )
             );
             return new HolderLookup.Provider() {
