--- a/net/minecraft/core/RegistrySetBuilder.java
+++ b/net/minecraft/core/RegistrySetBuilder.java
@@ -270,8 +270,8 @@
                 ResourceKey<?> resourcekey = entry.getKey();
                 if (resourcekey.isFor(this.key)) {
                     RegistrySetBuilder.RegisteredValue<T> registeredvalue = (RegistrySetBuilder.RegisteredValue)entry.getValue();
-                    Holder.Reference<T> reference = registries.lookup.holders.remove(resourcekey);
-                    map.put(resourcekey, new RegistrySetBuilder.ValueAndHolder<>(registeredvalue, Optional.ofNullable(reference)));
+                    Holder.Reference<T> reference = (Holder.Reference<T>) registries.lookup.holders.remove(resourcekey);
+                    map.put((ResourceKey<T>) resourcekey, new RegistrySetBuilder.ValueAndHolder<>(registeredvalue, Optional.ofNullable(reference)));
                     iterator.remove();
                 }
             }
@@ -293,7 +293,7 @@
         }
 
         <T> Holder.Reference<T> getOrCreate(ResourceKey<T> key) {
-            return this.holders.computeIfAbsent(key, key2 -> Holder.Reference.createStandAlone(this.owner, key2));
+            return (Holder.Reference<T>) this.holders.computeIfAbsent((ResourceKey<Object>) key, key2 -> Holder.Reference.createStandAlone(this.owner, key2));
         }
     }
 
