--- a/net/minecraft/world/item/BucketItem.java
+++ b/net/minecraft/world/item/BucketItem.java
@@ -99,16 +99,14 @@
 
     @Override
     public boolean emptyContents(@Nullable Player player, Level level, BlockPos pos, @Nullable BlockHitResult result) {
-        Fluid blockState = this.content;
-        if (!(blockState instanceof FlowingFluid)) {
+        if (!(this.content instanceof FlowingFluid flowingFluid)) {
             return false;
         } else {
-            FlowingFluid flowingFluid;
             Block block;
             boolean canBeReplaced;
             boolean var10000;
+            BlockState blockState;
             label82: {
-                flowingFluid = (FlowingFluid)blockState;
                 blockState = level.getBlockState(pos);
                 block = blockState.getBlock();
                 canBeReplaced = blockState.canBeReplaced(this.content);
