--- a/net/minecraft/commands/arguments/blocks/BlockPredicateArgument.java
+++ b/net/minecraft/commands/arguments/blocks/BlockPredicateArgument.java
@@ -85,7 +85,7 @@
                 return false;
             } else {
                 for (Property<?> property : this.properties) {
-                    if (state.getValue((Property<boolean>)property) != this.state.getValue((Property<boolean>)property)) {
+                    if (state.getValue(property) != this.state.getValue(property)) {
                         return false;
                     }
                 }
@@ -138,7 +138,7 @@
                         return false;
                     }
 
-                    if (state.getValue((Property<boolean>)property) != comparable) {
+                    if (state.getValue(property) != comparable) {
                         return false;
                     }
                 }
