--- a/net/minecraft/server/commands/data/DataCommands.java
+++ b/net/minecraft/server/commands/data/DataCommands.java
@@ -305,15 +305,15 @@
     }
 
     private static String substring(String string, int startIndex, int endIndex) throws CommandSyntaxException {
-        int len = string.length();
-        int offset = getOffset(startIndex, len);
-        int offset1 = getOffset(endIndex, len);
+        int i = string.length();
+        int offset = getOffset(startIndex, i);
+        int offset1 = getOffset(endIndex, i);
         return validatedSubstring(string, offset, offset1);
     }
 
     private static String substring(String string, int startIndex) throws CommandSyntaxException {
-        int len = string.length();
-        return validatedSubstring(string, getOffset(startIndex, len), len);
+        int i = string.length();
+        return validatedSubstring(string, getOffset(startIndex, i), i);
     }
 
     private static int getOffset(int index, int length) {
