--- a/net/minecraft/resources/RegistryOps.java
+++ b/net/minecraft/resources/RegistryOps.java
@@ -23,7 +23,7 @@
 
             @Override
             public <T> Optional<RegistryOps.RegistryInfo<T>> lookup(ResourceKey<? extends Registry<? extends T>> registryRef) {
-                return this.lookups.computeIfAbsent(registryRef, registryInfoGetter::lookup);
+                return (Optional<RegistryInfo<T>>) this.lookups.computeIfAbsent(registryRef, registryInfoGetter::lookup);
             }
         };
     }
