--- a/net/minecraft/world/entity/EntityReference.java
+++ b/net/minecraft/world/entity/EntityReference.java
@@ -93,7 +_,7 @@
 
     @Nullable
     public static <StoredEntityType extends UniquelyIdentifyable> EntityReference<StoredEntityType> read(ValueInput input, String key) {
-        return input.read(key, codec()).orElse(null);
+        return input.read(key, EntityReference.<StoredEntityType>codec()).orElse(null);
     }
 
     @Nullable
@@ -105,7 +_,7 @@
             ? new EntityReference<>(optional.get())
             : input.getString(key)
                 .map(string -> OldUsersConverter.convertMobOwnerIfNecessary(level.getServer(), string))
-                .map(EntityReference::new)
+                .map(EntityReference<StoredEntityType>::new)
                 .orElse(null);
     }
 
