--- a/net/minecraft/world/level/storage/loot/GsonAdapterFactory.java
+++ b/net/minecraft/world/level/storage/loot/GsonAdapterFactory.java
@@ -116,13 +116,13 @@
         public JsonElement serialize(E object, Type type, JsonSerializationContext context) {
             T t = this.typeGetter.apply(object);
             if (this.inlineType != null && this.inlineType.getFirst() == t) {
-                return this.inlineType.getSecond().serialize(object, context);
+                return ((InlineSerializer<E>) this.inlineType.getSecond()).serialize(object, context);
             } else if (t == null) {
                 throw new JsonSyntaxException("Unknown type: " + object);
             } else {
                 JsonObject jsonobject = new JsonObject();
                 jsonobject.addProperty(this.typeKey, this.registry.getKey(t).toString());
-                t.getSerializer().serialize(jsonobject, object, context);
+                ((Serializer<E>) t.getSerializer()).serialize(jsonobject, object, context);
                 return jsonobject;
             }
         }
