--- a/net/minecraft/world/phys/shapes/NonOverlappingMerger.java
+++ b/net/minecraft/world/phys/shapes/NonOverlappingMerger.java
@@ -25,18 +25,18 @@
     }
 
     private boolean forNonSwappedIndexes(IndexMerger.IndexConsumer predicate) {
-        int size = this.lower.size();
+        int i = this.lower.size();
 
-        for(int i = 0; i < size; ++i) {
-            if (!predicate.merge(i, -1, i)) {
+        for(int i1 = 0; i1 < i; ++i1) {
+            if (!predicate.merge(i1, -1, i1)) {
                 return false;
             }
         }
 
-        int i = this.upper.size() - 1;
+        int i1 = this.upper.size() - 1;
 
-        for(int i1 = 0; i1 < i; ++i1) {
-            if (!predicate.merge(size - 1, i1, size + i1)) {
+        for(int i2 = 0; i2 < i1; ++i2) {
+            if (!predicate.merge(i - 1, i2, i + i2)) {
                 return false;
             }
         }
