--- a/net/minecraft/util/EncoderCache.java
+++ b/net/minecraft/util/EncoderCache.java
@@ -36,7 +_,7 @@
             public <T> DataResult<T> encode(A input, DynamicOps<T> ops, T value) {
                 return EncoderCache.this.cache
                     .getUnchecked(new EncoderCache.Key<>(codec, input, ops))
-                    .map(object -> object instanceof Tag tag ? tag.copy() : object);
+                    .map(object -> (T) (object instanceof Tag tag ? tag.copy() : object));
             }
         };
     }
