--- a/net/minecraft/world/level/block/BeehiveBlock.java
+++ b/net/minecraft/world/level/block/BeehiveBlock.java
@@ -91,11 +91,11 @@
         List<Bee> entitiesOfClass = world.getEntitiesOfClass(Bee.class, new AABB(pos).inflate(8.0, 6.0, 8.0));
         if (!entitiesOfClass.isEmpty()) {
             List<Player> entitiesOfClass1 = world.getEntitiesOfClass(Player.class, new AABB(pos).inflate(8.0, 6.0, 8.0));
-            int size = entitiesOfClass1.size();
+            int i = entitiesOfClass1.size();
 
             for(Bee bee : entitiesOfClass) {
                 if (bee.getTarget() == null) {
-                    bee.setTarget(entitiesOfClass1.get(world.random.nextInt(size)));
+                    bee.setTarget(entitiesOfClass1.get(world.random.nextInt(i)));
                 }
             }
         }
