--- a/net/minecraft/world/level/block/Block.java
+++ b/net/minecraft/world/level/block/Block.java
@@ -524,9 +524,9 @@
 
         @Override
         public int hashCode() {
-            int hashCode = this.first.hashCode();
-            hashCode = 31 * hashCode + this.second.hashCode();
-            return 31 * hashCode + this.direction.hashCode();
+            int i = this.first.hashCode();
+            i = 31 * i + this.second.hashCode();
+            return 31 * i + this.direction.hashCode();
         }
     }
 }
