--- a/net/minecraft/util/ExtraCodecs.java
+++ b/net/minecraft/util/ExtraCodecs.java
@@ -218,7 +_,7 @@
                 P object1 = list1.get(1);
                 return factory.apply(object, object1);
             }), object -> ImmutableList.of(minGetter.apply((I)object), maxGetter.apply((I)object)));
-        Codec<I> codec2 = RecordCodecBuilder.<Pair>create(
+        Codec<I> codec2 = RecordCodecBuilder.<Pair<P, P>>create(
                 instance -> instance.group(codec.fieldOf(minFieldName).forGetter(Pair::getFirst), codec.fieldOf(maxFieldName).forGetter(Pair::getSecond))
                         .apply(instance, Pair::of)
             )
