--- a/net/minecraft/core/RegistrySetBuilder.java
+++ b/net/minecraft/core/RegistrySetBuilder.java
@@ -264,7 +264,7 @@
         }
 
         public <T> HolderOwner<T> cast() {
-            return this;
+            return (HolderOwner<T>) this;
         }
     }
 
@@ -379,7 +379,7 @@
 
         <T> Holder.Reference<T> getOrCreate(ResourceKey<T> key) {
             return (Holder.Reference<T>)this.holders
-                .computeIfAbsent(key, resourceKey -> Holder.Reference.createStandAlone(this.owner, (ResourceKey<Object>)resourceKey));
+                .computeIfAbsent((ResourceKey<Object>) key, resourceKey -> Holder.Reference.createStandAlone(this.owner, (ResourceKey<Object>)resourceKey));
         }
     }
 
