--- a/net/minecraft/world/level/gamerules/GameRuleMap.java
+++ b/net/minecraft/world/level/gamerules/GameRuleMap.java
@@ -12,7 +_,7 @@
 import org.jspecify.annotations.Nullable;
 
 public final class GameRuleMap {
-    public static final Codec<GameRuleMap> CODEC = Codec.dispatchedMap(BuiltInRegistries.GAME_RULE.byNameCodec(), GameRule::valueCodec)
+    public static final Codec<GameRuleMap> CODEC = Codec.<GameRule<?>, Object>dispatchedMap(BuiltInRegistries.GAME_RULE.byNameCodec(), GameRule::valueCodec)
         .xmap(GameRuleMap::ofTrusted, GameRuleMap::map);
     private final Reference2ObjectMap<GameRule<?>, Object> map;
 
