--- a/net/minecraft/world/item/enchantment/ConditionalEffect.java
+++ b/net/minecraft/world/item/enchantment/ConditionalEffect.java
@@ -19,7 +_,7 @@
                     ValidationContext validationContext = new ValidationContext(collector, lootContextParamSet);
                     lootItemCondition.validate(validationContext);
                     return collector.getReport()
-                        .map(string -> DataResult.error(() -> "Validation error in enchantment effect condition: " + string))
+                        .map(string -> DataResult.<LootItemCondition>error(() -> "Validation error in enchantment effect condition: " + string))
                         .orElseGet(() -> DataResult.success(lootItemCondition));
                 }
             );
