--- a/net/minecraft/world/level/storage/loot/GsonAdapterFactory.java
+++ b/net/minecraft/world/level/storage/loot/GsonAdapterFactory.java
@@ -116,13 +116,13 @@
         public JsonElement serialize(E object, Type type, JsonSerializationContext jsonSerializationContext) {
             T serializerType = this.typeGetter.apply(object);
             if (this.inlineType != null && this.inlineType.getFirst() == serializerType) {
-                return this.inlineType.getSecond().serialize(object, jsonSerializationContext);
+                return ((InlineSerializer<E>) this.inlineType.getSecond()).serialize(object, jsonSerializationContext);
             } else if (serializerType == null) {
                 throw new JsonSyntaxException("Unknown type: " + object);
             } else {
                 JsonObject jsonObject = new JsonObject();
                 jsonObject.addProperty(this.typeKey, this.registry.getKey(serializerType).toString());
-                serializerType.getSerializer().serialize(jsonObject, object, jsonSerializationContext);
+                ((Serializer<E>) serializerType.getSerializer()).serialize(jsonObject, object, jsonSerializationContext);
                 return jsonObject;
             }
         }
