--- a/net/minecraft/world/level/storage/loot/LootDataType.java
+++ b/net/minecraft/world/level/storage/loot/LootDataType.java
@@ -80,12 +80,12 @@
     private static <T> BiFunction<Gson, String, BiFunction<ResourceLocation, JsonElement, Optional<T>>> createSingleOrMultipleDeserialiser(
         Class<T> clazz1, Function<T[], T> combiner
     ) {
-        Class<T[]> clazz2 = clazz1.arrayType();
+        Class<T[]> clazz2 = (Class<T[]>) clazz1.arrayType();
         return (gson, dataTypeId) -> (id, json) -> {
                 try {
                     if (json.isJsonArray()) {
                         T[] objects = (T[])((Object[])gson.fromJson(json, clazz2));
-                        return Optional.of(combiner.apply((T)objects));
+                        return Optional.of(combiner.apply(objects));
                     } else {
                         return Optional.of(gson.fromJson(json, clazz1));
                     }
