--- a/net/minecraft/gametest/framework/GameTestRegistry.java
+++ b/net/minecraft/gametest/framework/GameTestRegistry.java
@@ -53,7 +53,7 @@
         T t = method.getAnnotation(clazz);
         if (t != null) {
             String s = batchIdFunction.apply(t);
-            Consumer<ServerLevel> consumer = batchConsumerMap.putIfAbsent(s, turnMethodIntoConsumer(method));
+            Consumer<ServerLevel> consumer = batchConsumerMap.putIfAbsent(s, (Consumer<ServerLevel>) turnMethodIntoConsumer(method));
             if (consumer != null) {
                 throw new RuntimeException("Hey, there should only be one " + clazz + " method per batch. Batch '" + s + "' has more than one!");
             }
@@ -126,7 +126,7 @@
             gametest.required(),
             gametest.requiredSuccesses(),
             gametest.attempts(),
-            turnMethodIntoConsumer(method)
+            (Consumer<GameTestHelper>) turnMethodIntoConsumer(method)
         );
     }
 
