--- a/net/minecraft/commands/Commands.java
+++ b/net/minecraft/commands/Commands.java
@@ -374,13 +_,13 @@
     ) {
         for (CommandNode<CommandSourceStack> commandNode : rootCommandSource.getChildren()) {
             if (commandNode.canUse(source)) {
-                ArgumentBuilder<SharedSuggestionProvider, ?> argumentBuilder = commandNode.createBuilder();
+                ArgumentBuilder<SharedSuggestionProvider, ?> argumentBuilder = (ArgumentBuilder) commandNode.createBuilder();
                 argumentBuilder.requires(suggestions -> true);
                 if (argumentBuilder.getCommand() != null) {
                     argumentBuilder.executes(commandContext -> 0);
                 }
 
-                if (argumentBuilder instanceof RequiredArgumentBuilder<SharedSuggestionProvider, ?> requiredArgumentBuilder
+                if (argumentBuilder instanceof RequiredArgumentBuilder requiredArgumentBuilder
                     && requiredArgumentBuilder.getSuggestionsProvider() != null) {
                     requiredArgumentBuilder.suggests(SuggestionProviders.safelySwap(requiredArgumentBuilder.getSuggestionsProvider()));
                 }
