--- a/net/minecraft/world/level/storage/loot/predicates/LootItemBlockStatePropertyCondition.java
+++ b/net/minecraft/world/level/storage/loot/predicates/LootItemBlockStatePropertyCondition.java
@@ -32,7 +32,7 @@
     private static DataResult<LootItemBlockStatePropertyCondition> validate(LootItemBlockStatePropertyCondition lootItemBlockStatePropertyCondition) {
         return lootItemBlockStatePropertyCondition.properties()
             .flatMap(statePropertiesPredicate -> statePropertiesPredicate.checkState(lootItemBlockStatePropertyCondition.block().value().getStateDefinition()))
-            .map(string -> DataResult.error(() -> "Block " + lootItemBlockStatePropertyCondition.block() + " has no property" + string))
+            .map(string -> DataResult.<LootItemBlockStatePropertyCondition>error(() -> "Block " + lootItemBlockStatePropertyCondition.block() + " has no property" + string))
             .orElse(DataResult.success(lootItemBlockStatePropertyCondition));
     }
 
