--- a/net/minecraft/world/level/block/state/StateHolder.java
+++ b/net/minecraft/world/level/block/state/StateHolder.java
@@ -132,7 +_,7 @@
 
             for (Entry<Property<?>, Comparable<?>> entry : this.values.entrySet()) {
                 Property<?> property = entry.getKey();
-                map.put(property, property.getPossibleValues().stream().map(comparable -> {
+                map.put(property, (S[]) property.getPossibleValues().stream().map(comparable -> {
                     return possibleStateMap.get(this.makeNeighbourValues(property, comparable));
                 }).toArray());
             }
