--- a/net/minecraft/util/ExtraCodecs.java
+++ b/net/minecraft/util/ExtraCodecs.java
@@ -276,7 +_,7 @@
             P max = l.get(1);
             return makeInterval.apply(min, max);
         }), p -> ImmutableList.of(getMin.apply((I)p), getMax.apply((I)p)));
-        Codec<I> objectCodec = RecordCodecBuilder.<Pair>create(
+        Codec<I> objectCodec = RecordCodecBuilder.<Pair<P, P>>create(
                 i -> i.group(pointCodec.fieldOf(lowerBoundName).forGetter(Pair::getFirst), pointCodec.fieldOf(upperBoundName).forGetter(Pair::getSecond))
                     .apply(i, Pair::of)
             )
