--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -273,7 +273,7 @@
             }
 
             FullChunkStatus fullStatus = visibleChunkIfPresent.getFullStatus();
-            string = string + '§' + fullStatus.ordinal() + fullStatus;
+            string = string + "§" + fullStatus.ordinal() + fullStatus;
             return string + "§r";
         }
     }
@@ -588,7 +588,7 @@
                 }
 
                 return isChunkDataValid;
-            })).thenApplyAsync(optional -> {
+            })).<Either<ChunkAccess, ChunkHolder.ChunkLoadingFailure>>thenApplyAsync(optional -> {
             this.level.getProfiler().incrementCounter("chunkLoad");
             if (optional.isPresent()) {
                 ChunkAccess chunkAccess = ChunkSerializer.read(this.level, this.poiManager, chunkPos, optional.get());
