--- a/net/minecraft/world/entity/ai/behavior/AssignProfessionFromJobSite.java
+++ b/net/minecraft/world/entity/ai/behavior/AssignProfessionFromJobSite.java
@@ -14,7 +14,7 @@
 public class AssignProfessionFromJobSite {
     public static BehaviorControl<Villager> create() {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.present(MemoryModuleType.POTENTIAL_JOB_SITE), context.registered(MemoryModuleType.JOB_SITE)
                     )
                     .apply(
