--- a/net/minecraft/world/entity/ai/behavior/CopyMemoryWithExpiry.java
+++ b/net/minecraft/world/entity/ai/behavior/CopyMemoryWithExpiry.java
@@ -12,7 +12,7 @@
         Predicate<E> predicate, MemoryModuleType<? extends T> sourceType, MemoryModuleType<T> targetType, UniformInt expiry
     ) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor>group(context.present(sourceType), context.absent(targetType))
+            context -> context.group(context.present(sourceType), context.absent(targetType))
                     .apply(context, (source, target) -> (world, entity, time) -> {
                             if (!predicate.test(entity)) {
                                 return false;
