--- a/net/minecraft/world/entity/ai/behavior/DismountOrSkipMounting.java
+++ b/net/minecraft/world/entity/ai/behavior/DismountOrSkipMounting.java
@@ -10,7 +10,7 @@
 public class DismountOrSkipMounting {
     public static <E extends LivingEntity> BehaviorControl<E> create(int i, BiPredicate<E, Entity> alternativeRideCondition) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor>group(context.registered(MemoryModuleType.RIDE_TARGET)).apply(context, rideTarget -> (world, entity, time) -> {
+            context -> context.group(context.registered(MemoryModuleType.RIDE_TARGET)).apply(context, rideTarget -> (world, entity, time) -> {
                         Entity vehicle = entity.getVehicle();
                         Entity entity1 = context.<Entity>tryGet(rideTarget).orElse(null);
                         if (vehicle == null && entity1 == null) {
