--- a/net/minecraft/world/entity/ai/behavior/GoToTargetLocation.java
+++ b/net/minecraft/world/entity/ai/behavior/GoToTargetLocation.java
@@ -19,7 +19,7 @@
 
     public static <E extends Mob> OneShot<E> create(MemoryModuleType<BlockPos> memoryModuleType, int completionRange, float speed) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.present(memoryModuleType),
                         context.absent(MemoryModuleType.ATTACK_TARGET),
                         context.absent(MemoryModuleType.WALK_TARGET),
