--- a/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
+++ b/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
@@ -13,7 +13,7 @@
 public class InsideBrownianWalk {
     public static BehaviorControl<PathfinderMob> create(float f) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor>group(context.absent(MemoryModuleType.WALK_TARGET))
+            context -> context.group(context.absent(MemoryModuleType.WALK_TARGET))
                     .apply(
                         context,
                         walkTarget -> (world, entity, time) -> {
