--- a/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
+++ b/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
@@ -34,7 +34,7 @@
         MutableObject<Node> mutableObject = new MutableObject<>(null);
         MutableInt mutableInt = new MutableInt(0);
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.present(MemoryModuleType.PATH),
                         context.registered(MemoryModuleType.DOORS_TO_CLOSE),
                         context.registered(MemoryModuleType.NEAREST_LIVING_ENTITIES)
@@ -159,7 +159,7 @@
         MemoryAccessor<Mu, Set<GlobalPos>> memoryAccessor, Optional<Set<GlobalPos>> doors, ServerLevel world, BlockPos pos
     ) {
         GlobalPos globalPos = GlobalPos.of(world.dimension(), pos);
-        return Optional.of(doors.map((Function<? super Set<GlobalPos>, ? extends Set<GlobalPos>>)(doorSet -> {
+        return Optional.of(doors.map((doorSet -> {
             doorSet.add(globalPos);
             return doorSet;
         })).orElseGet(() -> {
