--- a/net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.java
+++ b/net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.java
@@ -26,7 +26,7 @@
 
     public static BehaviorControl<PathfinderMob> create() {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.present(MemoryModuleType.VISIBLE_VILLAGER_BABIES),
                         context.absent(MemoryModuleType.WALK_TARGET),
                         context.registered(MemoryModuleType.LOOK_TARGET),
