--- a/net/minecraft/world/entity/ai/behavior/RingBell.java
+++ b/net/minecraft/world/entity/ai/behavior/RingBell.java
@@ -16,7 +16,7 @@
 
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor>group(context.present(MemoryModuleType.MEETING_POINT)).apply(context, meetingPoint -> (world, entity, time) -> {
+            context -> context.group(context.present(MemoryModuleType.MEETING_POINT)).apply(context, meetingPoint -> (world, entity, time) -> {
                         if (world.random.nextFloat() <= 0.95F) {
                             return false;
                         } else {
