--- a/net/minecraft/world/entity/ai/behavior/SetEntityLookTargetSometimes.java
+++ b/net/minecraft/world/entity/ai/behavior/SetEntityLookTargetSometimes.java
@@ -25,7 +25,7 @@
         float f1 = f * f;
         SetEntityLookTargetSometimes.Ticker ticker = new SetEntityLookTargetSometimes.Ticker(interval);
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.absent(MemoryModuleType.LOOK_TARGET), context.present(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES)
                     )
                     .apply(
