--- a/net/minecraft/world/entity/ai/behavior/SetLookAndInteract.java
+++ b/net/minecraft/world/entity/ai/behavior/SetLookAndInteract.java
@@ -12,7 +12,7 @@
     public static BehaviorControl<LivingEntity> create(EntityType<?> entityType, int maxDistance) {
         int i = maxDistance * maxDistance;
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.registered(MemoryModuleType.LOOK_TARGET),
                         context.absent(MemoryModuleType.INTERACTION_TARGET),
                         context.present(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES)
