--- a/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromAttackTargetIfTargetOutOfReach.java
+++ b/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromAttackTargetIfTargetOutOfReach.java
@@ -19,7 +19,7 @@
 
     public static BehaviorControl<Mob> create(Function<LivingEntity, Float> function) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.registered(MemoryModuleType.WALK_TARGET),
                         context.registered(MemoryModuleType.LOOK_TARGET),
                         context.present(MemoryModuleType.ATTACK_TARGET),
