--- a/net/minecraft/world/entity/ai/behavior/StartCelebratingIfTargetDead.java
+++ b/net/minecraft/world/entity/ai/behavior/StartCelebratingIfTargetDead.java
@@ -11,7 +11,7 @@
 public class StartCelebratingIfTargetDead {
     public static BehaviorControl<LivingEntity> create(int i, BiPredicate<LivingEntity, LivingEntity> predicate) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.present(MemoryModuleType.ATTACK_TARGET),
                         context.registered(MemoryModuleType.ANGRY_AT),
                         context.absent(MemoryModuleType.CELEBRATE_LOCATION),
