--- a/net/minecraft/world/entity/ai/behavior/StayCloseToTarget.java
+++ b/net/minecraft/world/entity/ai/behavior/StayCloseToTarget.java
@@ -14,7 +14,7 @@
         Function<LivingEntity, Optional<PositionTracker>> function, Predicate<LivingEntity> predicate, int completionRange, int searchRange, float speed
     ) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.registered(MemoryModuleType.LOOK_TARGET), context.registered(MemoryModuleType.WALK_TARGET)
                     )
                     .apply(context, (lookTarget, walkTarget) -> (world, entity, time) -> {
