--- a/net/minecraft/world/entity/ai/behavior/StopBeingAngryIfTargetDead.java
+++ b/net/minecraft/world/entity/ai/behavior/StopBeingAngryIfTargetDead.java
@@ -11,7 +11,7 @@
 public class StopBeingAngryIfTargetDead {
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor>group(instance.present(MemoryModuleType.ANGRY_AT))
+            instance -> instance.group(instance.present(MemoryModuleType.ANGRY_AT))
                     .apply(
                         instance,
                         angryAt -> (world, entity, time) -> {
