--- a/net/minecraft/world/entity/ai/behavior/TryLaySpawnOnWaterNearLand.java
+++ b/net/minecraft/world/entity/ai/behavior/TryLaySpawnOnWaterNearLand.java
@@ -16,7 +16,7 @@
 public class TryLaySpawnOnWaterNearLand {
     public static BehaviorControl<LivingEntity> create(Block block) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.absent(MemoryModuleType.ATTACK_TARGET),
                         context.present(MemoryModuleType.WALK_TARGET),
                         context.present(MemoryModuleType.IS_PREGNANT)
