--- a/net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.java
@@ -20,7 +20,7 @@
 
     public static BehaviorControl<LivingEntity> create(Predicate<Holder<PoiType>> predicate, MemoryModuleType<GlobalPos> poiPosModule) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor>group(context.present(poiPosModule)).apply(context, poiPos -> (world, entity, time) -> {
+            context -> context.group(context.present(poiPosModule)).apply(context, poiPos -> (world, entity, time) -> {
                         GlobalPos globalPos = context.get(poiPos);
                         BlockPos blockPos = globalPos.pos();
                         if (world.dimension() == globalPos.dimension() && blockPos.closerToCenterThan(entity.position(), 16.0)) {
