--- a/net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
+++ b/net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
@@ -22,7 +22,7 @@
 
     public static OneShot<PathfinderMob> create(float f, int horizontalRange, int verticalRange) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor>group(context.absent(MemoryModuleType.WALK_TARGET))
+            context -> context.group(context.absent(MemoryModuleType.WALK_TARGET))
                     .apply(
                         context,
                         walkTarget -> (world, entity, time) -> {
