--- a/net/minecraft/world/entity/ai/behavior/YieldJobSite.java
+++ b/net/minecraft/world/entity/ai/behavior/YieldJobSite.java
@@ -19,7 +19,7 @@
 public class YieldJobSite {
     public static BehaviorControl<Villager> create(float f) {
         return BehaviorBuilder.create(
-            context -> context.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            context -> context.group(
                         context.present(MemoryModuleType.POTENTIAL_JOB_SITE),
                         context.absent(MemoryModuleType.JOB_SITE),
                         context.present(MemoryModuleType.NEAREST_LIVING_ENTITIES),
