--- a/net/minecraft/data/loot/packs/VanillaBlockLoot.java
+++ b/net/minecraft/data/loot/packs/VanillaBlockLoot.java
@@ -802,7 +802,7 @@
                             .add(
                                 this.applyExplosionDecay(
                                     block,
-                                    (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.COCOA_BEANS)
+                                    LootItem.lootTableItem(Items.COCOA_BEANS)
                                         .apply(
                                             SetItemCountFunction.setCount(ConstantValue.exactly(3.0F))
                                                 .when(
@@ -823,7 +823,7 @@
                             .add(
                                 this.applyExplosionDecay(
                                     Blocks.SEA_PICKLE,
-                                    (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(block)
+                                    LootItem.lootTableItem(block)
                                         .apply(
                                             List.of(2, 3, 4),
                                             integer -> SetItemCountFunction.setCount(ConstantValue.exactly((float)integer.intValue()))
@@ -846,7 +846,7 @@
                         LootPool.lootPool()
                             .add(
                                 this.applyExplosionDecay(
-                                    block, (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.COMPOSTER)
+                                    block, LootItem.lootTableItem(Items.COMPOSTER)
                                 )
                             )
                     )
@@ -1094,7 +1094,7 @@
                     block,
                     this.applyExplosionDecay(
                         block,
-                        (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.GOLD_NUGGET)
+                        LootItem.lootTableItem(Items.GOLD_NUGGET)
                             .apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 6.0F)))
                             .apply(ApplyBonusCount.addOreBonusCount(Enchantments.BLOCK_FORTUNE))
                     )
@@ -1106,7 +1106,7 @@
             Blocks.COBWEB,
             block -> createSilkTouchOrShearsDispatchTable(
                     block,
-                    this.applyExplosionCondition(block, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.STRING))
+                    this.applyExplosionCondition(block, LootItem.lootTableItem(Items.STRING))
                 )
         );
         this.add(
@@ -1115,7 +1115,7 @@
                     block,
                     this.applyExplosionDecay(
                         block,
-                        (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.STICK)
+                        LootItem.lootTableItem(Items.STICK)
                             .apply(SetItemCountFunction.setCount(UniformGenerator.between(0.0F, 2.0F)))
                     )
                 )
@@ -1141,7 +1141,7 @@
                         LootPool.lootPool()
                             .setRolls(ConstantValue.exactly(1.0F))
                             .add(
-                                this.applyExplosionCondition(block, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(block))
+                                this.applyExplosionCondition(block, LootItem.lootTableItem(block))
                                     .when(LootItemEntityPropertyCondition.entityPresent(LootContext.EntityTarget.THIS))
                             )
                     )
@@ -1154,7 +1154,7 @@
                     block,
                     this.applyExplosionDecay(
                         block,
-                        (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.GLOWSTONE_DUST)
+                        LootItem.lootTableItem(Items.GLOWSTONE_DUST)
                             .apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 4.0F)))
                             .apply(ApplyBonusCount.addUniformBonusCount(Enchantments.BLOCK_FORTUNE))
                             .apply(LimitCount.limitCount(IntRange.range(1, 4)))
@@ -1167,7 +1167,7 @@
                     block,
                     this.applyExplosionDecay(
                         block,
-                        (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.MELON_SLICE)
+                        LootItem.lootTableItem(Items.MELON_SLICE)
                             .apply(SetItemCountFunction.setCount(UniformGenerator.between(3.0F, 7.0F)))
                             .apply(ApplyBonusCount.addUniformBonusCount(Enchantments.BLOCK_FORTUNE))
                             .apply(LimitCount.limitCount(IntRange.upperBound(9)))
@@ -1182,7 +1182,7 @@
                     block,
                     this.applyExplosionDecay(
                         block,
-                        (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.PRISMARINE_CRYSTALS)
+                        LootItem.lootTableItem(Items.PRISMARINE_CRYSTALS)
                             .apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 3.0F)))
                             .apply(ApplyBonusCount.addUniformBonusCount(Enchantments.BLOCK_FORTUNE))
                             .apply(LimitCount.limitCount(IntRange.range(1, 5)))
@@ -1274,7 +1274,7 @@
                     block,
                     this.applyExplosionCondition(
                         block,
-                        (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.CHARCOAL)
+                        LootItem.lootTableItem(Items.CHARCOAL)
                             .apply(SetItemCountFunction.setCount(ConstantValue.exactly(2.0F)))
                     )
                 )
@@ -1298,7 +1298,7 @@
                     block,
                     this.applyExplosionCondition(
                         block,
-                        (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.SOUL_SOIL)
+                        LootItem.lootTableItem(Items.SOUL_SOIL)
                             .apply(SetItemCountFunction.setCount(ConstantValue.exactly(1.0F)))
                     )
                 )
@@ -1314,7 +1314,7 @@
                         .otherwise(
                             this.applyExplosionDecay(
                                 block,
-                                (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.lootTableItem(Items.AMETHYST_SHARD)
+                                LootItem.lootTableItem(Items.AMETHYST_SHARD)
                                     .apply(SetItemCountFunction.setCount(ConstantValue.exactly(2.0F)))
                             )
                         )
