--- a/net/minecraft/server/packs/linkfs/LinkFSProvider.java
+++ b/net/minecraft/server/packs/linkfs/LinkFSProvider.java
@@ -85,7 +85,7 @@
                         } catch (IOException var3) {
                             throw new DirectoryIteratorException(var3);
                         }
-                    }).map(child -> child).iterator();
+                    }).map(child -> (Path) child).iterator();
                 }
 
                 @Override
