--- a/net/minecraft/world/entity/ai/behavior/StrollToPoiList.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollToPoiList.java
@@ -20,7 +20,7 @@
     ) {
         MutableLong mutableLong = new MutableLong(0L);
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.registered(MemoryModuleType.WALK_TARGET), instance.present(poiListMemory), instance.present(mustBeCloseToMemory)
                     )
                     .apply(
