--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -1123,7 +1123,7 @@
                 if (this.level() instanceof ServerLevel) {
                     SpawnEggItem spawneggitem = (SpawnEggItem)itemstack.getItem();
                     Optional<Mob> optional = spawneggitem.spawnOffspringFromSpawnEgg(
-                        player, this, this.getType(), (ServerLevel)this.level(), this.position(), itemstack
+                        player, this, (EntityType<? extends Mob>) this.getType(), (ServerLevel)this.level(), this.position(), itemstack
                     );
                     optional.ifPresent(entity -> this.onOffspringSpawnedFromEgg(player, entity));
                     return optional.isPresent() ? InteractionResult.SUCCESS : InteractionResult.PASS;
