--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -1123,7 +1123,7 @@
                 if (this.level() instanceof ServerLevel) {
                     SpawnEggItem spawnEggItem = (SpawnEggItem)itemInHand.getItem();
                     Optional<Mob> optional = spawnEggItem.spawnOffspringFromSpawnEgg(
-                        player, this, this.getType(), (ServerLevel)this.level(), this.position(), itemInHand
+                        player, this, (EntityType<Mob>) this.getType(), (ServerLevel)this.level(), this.position(), itemInHand
                     );
                     optional.ifPresent(mob -> this.onOffspringSpawnedFromEgg(player, mob));
                     return optional.isPresent() ? InteractionResult.SUCCESS : InteractionResult.PASS;
