--- a/net/minecraft/world/entity/SpawnPlacements.java
+++ b/net/minecraft/world/entity/SpawnPlacements.java
@@ -65,7 +65,7 @@
         EntityType<T> type, ServerLevelAccessor world, MobSpawnType spawnReason, BlockPos pos, RandomSource random
     ) {
         SpawnPlacements.Data spawnplacements$data = DATA_BY_TYPE.get(type);
-        return spawnplacements$data == null || spawnplacements$data.predicate.test(type, world, spawnReason, pos, random);
+        return spawnplacements$data == null || spawnplacements$data.predicate.test((EntityType) type, world, spawnReason, pos, random);
     }
 
     static {
