--- a/net/minecraft/core/Registry.java
+++ b/net/minecraft/core/Registry.java
@@ -152,7 +_,7 @@
     Optional<HolderSet.Named<T>> getTag(TagKey<T> key);
 
     default Iterable<Holder<T>> getTagOrEmpty(TagKey<T> key) {
-        return DataFixUtils.orElse(this.getTag(key), List.<T>of());
+        return DataFixUtils.orElse(this.getTag(key), List.of());
     }
 
     default Optional<Holder<T>> getRandomElementOf(TagKey<T> key, RandomSource random) {
