--- a/net/minecraft/world/entity/ai/behavior/warden/SetWardenLookTarget.java
+++ b/net/minecraft/world/entity/ai/behavior/warden/SetWardenLookTarget.java
@@ -13,7 +13,7 @@
 public class SetWardenLookTarget {
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.registered(MemoryModuleType.LOOK_TARGET),
                         instance.registered(MemoryModuleType.DISTURBANCE_LOCATION),
                         instance.registered(MemoryModuleType.ROAR_TARGET),
