--- a/net/minecraft/world/flag/FeatureFlags.java
+++ b/net/minecraft/world/flag/FeatureFlags.java
@@ -9,6 +9,12 @@
     public static final FeatureFlag VANILLA;
     public static final FeatureFlag BUNDLE;
     public static final FeatureFlagRegistry REGISTRY;
+    static {
+        FeatureFlagRegistry.Builder builder = new FeatureFlagRegistry.Builder("main");
+        VANILLA = builder.createVanilla("vanilla");
+        BUNDLE = builder.createVanilla("bundle");
+        REGISTRY = builder.build();
+    }
     public static final Codec<FeatureFlagSet> CODEC = REGISTRY.codec();
     public static final FeatureFlagSet VANILLA_SET = FeatureFlagSet.of(VANILLA);
     public static final FeatureFlagSet DEFAULT_FLAGS = VANILLA_SET;
@@ -26,11 +32,4 @@
     public static boolean isExperimental(FeatureFlagSet features) {
         return !features.isSubsetOf(VANILLA_SET);
     }
-
-    static {
-        FeatureFlagRegistry.Builder builder = new FeatureFlagRegistry.Builder("main");
-        VANILLA = builder.createVanilla("vanilla");
-        BUNDLE = builder.createVanilla("bundle");
-        REGISTRY = builder.build();
-    }
 }
