--- a/net/minecraft/world/flag/FeatureFlags.java
+++ b/net/minecraft/world/flag/FeatureFlags.java
@@ -9,9 +9,9 @@
     public static final FeatureFlag VANILLA;
     public static final FeatureFlag BUNDLE;
     public static final FeatureFlagRegistry REGISTRY;
-    public static final Codec<FeatureFlagSet> CODEC = REGISTRY.codec();
-    public static final FeatureFlagSet VANILLA_SET = FeatureFlagSet.of(VANILLA);
-    public static final FeatureFlagSet DEFAULT_FLAGS = VANILLA_SET;
+    public static final Codec<FeatureFlagSet> CODEC;
+    public static final FeatureFlagSet VANILLA_SET;
+    public static final FeatureFlagSet DEFAULT_FLAGS;
 
     public static String printMissingFlags(FeatureFlagSet enabledFeatures, FeatureFlagSet requestedFeatures) {
         return printMissingFlags(REGISTRY, enabledFeatures, requestedFeatures);
@@ -32,5 +32,8 @@
         VANILLA = builder.createVanilla("vanilla");
         BUNDLE = builder.createVanilla("bundle");
         REGISTRY = builder.build();
+        CODEC = REGISTRY.codec();
+        VANILLA_SET = FeatureFlagSet.of(VANILLA);
+        DEFAULT_FLAGS = VANILLA_SET;
     }
 }
