--- a/net/minecraft/world/entity/EntityReference.java
+++ b/net/minecraft/world/entity/EntityReference.java
@@ -118,7 +_,7 @@
     }
 
     public static <StoredEntityType extends UniquelyIdentifyable> @Nullable EntityReference<StoredEntityType> read(final ValueInput input, final String key) {
-        return input.read(key, codec()).orElse(null);
+        return (EntityReference<StoredEntityType>) input.read(key, codec()).orElse(null);
     }
 
     public static <StoredEntityType extends UniquelyIdentifyable> @Nullable EntityReference<StoredEntityType> readWithOldOwnerConversion(
@@ -127,7 +_,7 @@
         Optional<UUID> uuid = input.read(key, UUIDUtil.CODEC);
         return uuid.isPresent()
             ? of(uuid.get())
-            : input.getString(key)
+            : (EntityReference<StoredEntityType>) input.getString(key)
                 .map(oldName -> OldUsersConverter.convertMobOwnerIfNecessary(level.getServer(), oldName))
                 .map(EntityReference::new)
                 .orElse(null);
