--- a/net/minecraft/world/level/block/entity/SkullBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/SkullBlockEntity.java
@@ -179,7 +179,7 @@
             ? CompletableFuture.completedFuture(Optional.empty())
             : gameProfileCache.getAsync(string)
                 .thenCompose(optional -> optional.isPresent() ? fillProfileTextures(optional.get()) : CompletableFuture.completedFuture(Optional.empty()))
-                .thenApplyAsync((Function<? super Optional, ? extends Optional<GameProfile>>)(optional -> {
+                .thenApplyAsync((optional -> {
                     GameProfileCache gameProfileCache1 = profileCache;
                     if (gameProfileCache1 != null) {
                         optional.ifPresent(gameProfileCache1::add);
