--- a/net/minecraft/world/entity/EntityReference.java
+++ b/net/minecraft/world/entity/EntityReference.java
@@ -86,7 +_,7 @@
 
     @Nullable
     public static <StoredEntityType extends UniquelyIdentifyable> EntityReference<StoredEntityType> read(CompoundTag tag, String key) {
-        return tag.read(key, codec()).orElse(null);
+        return tag.read(key, EntityReference.<StoredEntityType>codec()).orElse(null);
     }
 
     @Nullable
@@ -96,6 +_,6 @@
         Optional<UUID> optional = tag.read(key, UUIDUtil.CODEC);
         return optional.isPresent()
             ? new EntityReference<>(optional.get())
-            : tag.getString(key).map(string -> OldUsersConverter.convertMobOwnerIfNecessary(level.theGame(), string)).map(EntityReference::new).orElse(null);
+            : tag.getString(key).map(string -> OldUsersConverter.convertMobOwnerIfNecessary(level.theGame(), string)).map(EntityReference<StoredEntityType>::new).orElse(null);
     }
 }
