--- a/net/minecraft/commands/Commands.java
+++ b/net/minecraft/commands/Commands.java
@@ -372,7 +372,7 @@
     ) {
         for (CommandNode<CommandSourceStack> commandNode : rootCommandSource.getChildren()) {
             if (commandNode.canUse(source)) {
-                ArgumentBuilder<SharedSuggestionProvider, ?> argumentBuilder = commandNode.createBuilder();
+                ArgumentBuilder<SharedSuggestionProvider, ?> argumentBuilder = (ArgumentBuilder) commandNode.createBuilder();
                 argumentBuilder.requires(sharedSuggestionProvider -> true);
                 if (argumentBuilder.getCommand() != null) {
                     argumentBuilder.executes(commandContext -> 0);
