--- a/net/minecraft/commands/functions/MacroFunction.java
+++ b/net/minecraft/commands/functions/MacroFunction.java
@@ -86,46 +_,11 @@
         Objects.requireNonNull(tag);
 
         return switch (tag) {
-            case FloatTag(float var41) -> {
-                float var25 = var41;
-                if (true) {
-                    yield DECIMAL_FORMAT.format(var25);
-                }
-
-                byte var45 = 1;
-            }
-            case DoubleTag(double var39) -> {
-                double var26 = var39;
-                if (true) {
-                    yield DECIMAL_FORMAT.format(var26);
-                }
-
-                byte var44 = 2;
-            }
-            case ByteTag(byte var37) -> {
-                byte var27 = var37;
-                if (true) {
-                    yield String.valueOf((int)var27);
-                }
-
-                byte var43 = 3;
-            }
-            case ShortTag(short var35) -> {
-                short var28 = var35;
-                if (true) {
-                    yield String.valueOf((int)var28);
-                }
-
-                byte var42 = 4;
-            }
-            case LongTag(long var33) -> {
-                long var29 = var33;
-                if (true) {
-                    yield String.valueOf(var29);
-                }
-
-                byte var2 = 5;
-            }
+            case FloatTag(float value) -> DECIMAL_FORMAT.format(value);
+            case DoubleTag(double value) -> DECIMAL_FORMAT.format(value);
+            case ByteTag(byte value) -> String.valueOf(value);
+            case ShortTag(short value) -> String.valueOf(value);
+            case LongTag(long value) -> String.valueOf(value);
             case StringTag(String var17) -> var17;
             default -> tag.toString();
         };
