--- a/net/minecraft/ChatFormatting.java
+++ b/net/minecraft/ChatFormatting.java
@@ -37,7 +37,7 @@
     RESET("RESET", 'r', -1, null);
 
     public static final Codec<ChatFormatting> CODEC = StringRepresentable.fromEnum(ChatFormatting::values);
-    public static final char PREFIX_CODE = '§';
+    public static final char PREFIX_CODE = '\u00A7';
     private static final Map<String, ChatFormatting> FORMATTING_BY_NAME = Arrays.stream(values())
         .collect(Collectors.toMap(formatting -> cleanName(formatting.name), chatFormatter -> chatFormatter));
     private static final Pattern STRIP_FORMATTING_PATTERN = Pattern.compile("(?i)§[0-9A-FK-OR]");
