--- a/net/minecraft/world/inventory/RecipeBookMenu.java
+++ b/net/minecraft/world/inventory/RecipeBookMenu.java
@@ -12,7 +12,7 @@
     }
 
     public void handlePlacement(boolean placeAll, Recipe<?> recipe, ServerPlayer player) {
-        new ServerPlaceRecipe<>(this).recipeClicked(player, recipe, placeAll);
+        new ServerPlaceRecipe<>(this).recipeClicked(player, (Recipe<C>) recipe, placeAll);
     }
 
     public abstract void fillCraftSlotsStackedContents(StackedContents itemHelper);
