--- a/net/minecraft/world/entity/animal/happyghast/HappyGhastAi.java
+++ b/net/minecraft/world/entity/animal/happyghast/HappyGhastAi.java
@@ -51,7 +_,7 @@
             0,
             ImmutableList.of(
                 new Swim<>(0.8F),
-                new AnimalPanic(2.0F, 0),
+                new AnimalPanic<>(2.0F, 0),
                 new LookAtTargetSink(45, 90),
                 new MoveToTargetSink(),
                 new CountDownCooldownTicks(MemoryModuleType.TEMPTATION_COOLDOWN_TICKS)
