--- a/net/minecraft/world/level/GameRules.java
+++ b/net/minecraft/world/level/GameRules.java
@@ -288,8 +_,8 @@
 
     private <T extends GameRules.Value<T>> void callVisitorCap(GameRules.GameRuleTypeVisitor visitor, GameRules.Key<?> key, GameRules.Type<?> type) {
         if (type.requiredFeatures.isSubsetOf(this.enabledFeatures)) {
-            visitor.visit(key, type);
-            type.callVisitor(visitor, key);
+            visitor.visit((GameRules.Key<T>) key, (GameRules.Type<T>) type);
+            ((GameRules.Type<T>) type).callVisitor(visitor, (GameRules.Key<T>) key);
         }
     }
 
