--- a/net/minecraft/world/entity/ai/gossip/GossipContainer.java
+++ b/net/minecraft/world/entity/ai/gossip/GossipContainer.java
@@ -159,7 +159,7 @@
             .decode(dynamic)
             .resultOrPartial(string -> LOGGER.warn("Failed to deserialize gossips: {}", string))
             .stream()
-            .flatMap(pair -> ((List)pair.getFirst()).stream())
+            .flatMap(pair -> (pair.getFirst()).stream())
             .forEach(gossipEntry -> this.getOrCreate(gossipEntry.target).entries.put(gossipEntry.type, gossipEntry.value));
     }
 
