--- a/net/minecraft/core/Registry.java
+++ b/net/minecraft/core/Registry.java
@@ -150,7 +150,7 @@
     Optional<HolderSet.Named<T>> getTag(TagKey<T> key);
 
     default Iterable<Holder<T>> getTagOrEmpty(TagKey<T> key) {
-        return DataFixUtils.orElse(this.getTag(key), List.<T>of());
+        return DataFixUtils.orElse(this.getTag(key), List.of());
     }
 
     HolderSet.Named<T> getOrCreateTag(TagKey<T> key);
@@ -173,7 +173,7 @@
             @Nullable
             @Override
             public Holder<T> byId(int id) {
-                return (Holder<T>)Registry.this.getHolder(id).orElse((T)null);
+                return (Holder<T>)Registry.this.getHolder(id).orElse(null);
             }
 
             @Override
@@ -183,7 +183,7 @@
 
             @Override
             public Iterator<Holder<T>> iterator() {
-                return Registry.this.holders().map(reference -> reference).iterator();
+                return Registry.this.holders().map(reference -> (Holder<T>) reference).iterator();
             }
         };
     }
