--- a/net/minecraft/world/level/GameRules.java
+++ b/net/minecraft/world/level/GameRules.java
@@ -230,7 +_,7 @@
                     .filter(entry -> entry.getValue().valueClass == clazz)
                     .map(Entry::getKey)
                     .filter(key -> key.getId().equals(string))
-                    .map(key -> (GameRules.Key)key)
+                    .map(key -> (GameRules.Key<T>)key)
                     .findFirst()
                     .map(DataResult::success)
                     .orElseGet(() -> DataResult.error(() -> "Invalid game rule ID for type: " + string)),
@@ -303,8 +_,8 @@
 
     private <T extends GameRules.Value<T>> void callVisitorCap(GameRules.GameRuleTypeVisitor visitor, GameRules.Key<?> key, GameRules.Type<?> type) {
         if (type.requiredFeatures.isSubsetOf(this.enabledFeatures)) {
-            visitor.visit(key, type);
-            type.callVisitor(visitor, key);
+            visitor.visit((GameRules.Key<T>) key, (GameRules.Type<T>) type);
+            ((GameRules.Type<T>) type).callVisitor(visitor, (GameRules.Key<T>) key);
         }
     }
 
