--- a/net/minecraft/nbt/NbtUtils.java
+++ b/net/minecraft/nbt/NbtUtils.java
@@ -372,7 +_,7 @@
                     indent(indentLevel + 1, stringBuilder);
                 }
 
-                int ix = list.stream().mapToInt(String::length).max().orElse(0);
+                ix = list.stream().mapToInt(String::length).max().orElse(0);
                 String stringx = Strings.repeat(" ", ix);
 
                 for (int i2 = 0; i2 < list.size(); i2++) {
@@ -399,7 +_,7 @@
             case 11:
                 IntArrayTag intArrayTag = (IntArrayTag)tag;
                 int[] asIntArray = intArrayTag.getAsIntArray();
-                int ix = 0;
+                ix = 0;
 
                 for (int i3 : asIntArray) {
                     ix = Math.max(ix, String.format(Locale.ROOT, "%X", i3).length());
