--- a/net/minecraft/world/entity/monster/warden/Warden.java
+++ b/net/minecraft/world/entity/monster/warden/Warden.java
@@ -380,7 +_,7 @@
 
     @Override
     public Brain<Warden> getBrain() {
-        return super.getBrain();
+        return (Brain<Warden>) super.getBrain();
     }
 
     @Override
@@ -552,7 +_,7 @@
                 this.nodeEvaluator = new WalkNodeEvaluator();
                 return new PathFinder(this.nodeEvaluator, maxVisitedNodes) {
                     {
-                        Objects.requireNonNull(<VAR_NAMELESS_ENCLOSURE>);
+                        Objects.requireNonNull(nodeEvaluator);
                     }
 
                     @Override
