--- a/net/minecraft/network/syncher/EntityDataSerializer.java
+++ b/net/minecraft/network/syncher/EntityDataSerializer.java
@@ -19,12 +19,12 @@
         return new EntityDataSerializer.ForValueType<T>() {
             @Override
             public void write(FriendlyByteBuf friendlyByteBuf, T object) {
-                valueWriter.accept((T)friendlyByteBuf, object);
+                valueWriter.accept(friendlyByteBuf, object);
             }
 
             @Override
             public T read(FriendlyByteBuf friendlyByteBuf) {
-                return valueReader.apply((T)friendlyByteBuf);
+                return valueReader.apply(friendlyByteBuf);
             }
         };
     }
