--- a/net/minecraft/commands/arguments/NbtPathArgument.java
+++ b/net/minecraft/commands/arguments/NbtPathArgument.java
@@ -299,7 +299,7 @@
 
         @Override
         public void getTag(Tag tag, List<Tag> tags) {
-            if (tag instanceof CollectionTag collectionTag) {
+            if (tag instanceof CollectionTag<?> collectionTag) {
                 int size = collectionTag.size();
                 int i = this.index < 0 ? size + this.index : this.index;
                 if (0 <= i && i < size) {
@@ -320,7 +320,7 @@
 
         @Override
         public int setTag(Tag tag, Supplier<Tag> supplier) {
-            if (tag instanceof CollectionTag collectionTag) {
+            if (tag instanceof CollectionTag<?> collectionTag) {
                 int size = collectionTag.size();
                 int i = this.index < 0 ? size + this.index : this.index;
                 if (0 <= i && i < size) {
