--- a/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
+++ b/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
@@ -35,7 +35,7 @@
         MutableObject<Node> mutableObject = new MutableObject<>(null);
         MutableInt mutableInt = new MutableInt(0);
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.PATH),
                         instance.registered(MemoryModuleType.DOORS_TO_CLOSE),
                         instance.registered(MemoryModuleType.NEAREST_LIVING_ENTITIES)
@@ -162,7 +162,7 @@
         MemoryAccessor<Mu, Set<GlobalPos>> doorsToClose, Optional<Set<GlobalPos>> doorPositions, ServerLevel level, BlockPos pos
     ) {
         GlobalPos globalPos = GlobalPos.of(level.dimension(), pos);
-        return Optional.of(doorPositions.map((Function<? super Set<GlobalPos>, ? extends Set<GlobalPos>>)(set -> {
+        return Optional.of(doorPositions.map((set -> {
             set.add(globalPos);
             return set;
         })).orElseGet(() -> {
