--- a/net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.java
+++ b/net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.java
@@ -27,7 +27,7 @@
 
     public static BehaviorControl<PathfinderMob> create() {
         return BehaviorBuilder.create(
-            instance -> instance.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            instance -> instance.group(
                         instance.present(MemoryModuleType.VISIBLE_VILLAGER_BABIES),
                         instance.absent(MemoryModuleType.WALK_TARGET),
                         instance.registered(MemoryModuleType.LOOK_TARGET),
