--- a/net/minecraft/Util.java
+++ b/net/minecraft/Util.java
@@ -565,7 +_,7 @@
     public static <K extends Enum<K>, V> EnumMap<K, V> makeEnumMap(Class<K> clazz, Function<K, V> function) {
         EnumMap<K, V> map = new EnumMap<>(clazz);
 
-        for (K _enum : (Enum[])clazz.getEnumConstants()) {
+        for (K _enum : clazz.getEnumConstants()) {
             map.put(_enum, function.apply(_enum));
         }
 
