--- a/net/minecraft/network/protocol/game/ClientboundUpdateRecipesPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundUpdateRecipesPacket.java
@@ -48,6 +48,6 @@
     public static <T extends Recipe<?>> void toNetwork(FriendlyByteBuf friendlyByteBuf, RecipeHolder<?> recipeHolder) {
         friendlyByteBuf.writeResourceLocation(BuiltInRegistries.RECIPE_SERIALIZER.getKey(recipeHolder.value().getSerializer()));
         friendlyByteBuf.writeResourceLocation(recipeHolder.id());
-        recipeHolder.value().getSerializer().toNetwork(friendlyByteBuf, recipeHolder.value());
+        ((net.minecraft.world.item.crafting.RecipeSerializer<T>) recipeHolder.value().getSerializer()).toNetwork(friendlyByteBuf, (T) recipeHolder.value());
     }
 }
