--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -507,9 +507,10 @@
 
         int i1 = Math.max(0, this.unloadQueue.size() - 2000);
 
-        while((booleanSupplier.getAsBoolean() || i1 > 0) && (l = (long)this.unloadQueue.poll()) != null) {
+        Runnable runnable;
+        while((booleanSupplier.getAsBoolean() || i1 > 0) && (runnable = this.unloadQueue.poll()) != null) {
             --i1;
-            l.run();
+            runnable.run();
         }
 
         int i2 = 0;
@@ -594,7 +595,7 @@
                 }
 
                 return flag;
-            })).thenApplyAsync(optional -> {
+            })).<Either<ChunkAccess, ChunkHolder.ChunkLoadingFailure>>thenApplyAsync(optional -> {
             this.level.getProfiler().incrementCounter("chunkLoad");
             if (optional.isPresent()) {
                 ChunkAccess chunkAccess = ChunkSerializer.read(this.level, this.poiManager, chunkPos, optional.get());
