--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -1511,7 +1511,7 @@
 
             return object2IntOpenHashMap.object2IntEntrySet()
                 .stream()
-                .sorted(Comparator.comparing(Entry::getIntValue).reversed())
+                .sorted(Comparator.<Entry<String>, Integer>comparing(Entry::getIntValue).reversed())
                 .limit(5L)
                 .map(entry -> (String)entry.getKey() + ":" + entry.getIntValue())
                 .collect(Collectors.joining(","));
