--- a/net/minecraft/server/level/ServerEntity.java
+++ b/net/minecraft/server/level/ServerEntity.java
@@ -91,8 +91,7 @@
             this.lastPassengers = passengers;
         }
 
-        Entity i1 = this.entity;
-        if (i1 instanceof ItemFrame itemFrame && this.tickCount % 10 == 0) {
+        if (this.entity instanceof ItemFrame itemFrame && this.tickCount % 10 == 0) {
             ItemStack itemStack = itemFrame.getItem();
             if (itemStack.getItem() instanceof MapItem) {
                 Integer mapId = MapItem.getMapId(itemStack);
