--- a/net/minecraft/server/rcon/thread/RconClient.java
+++ b/net/minecraft/server/rcon/thread/RconClient.java
@@ -123,14 +123,14 @@
     }
 
     private void sendCmdResponse(int sessionToken, String message) throws IOException {
-        int len = message.length();
+        int i = message.length();
 
         do {
-            int i = 4096 <= len ? 4096 : len;
-            this.send(sessionToken, 0, message.substring(0, i));
-            message = message.substring(i);
-            len = message.length();
-        } while(0 != len);
+            int i1 = 4096 <= i ? 4096 : i;
+            this.send(sessionToken, 0, message.substring(0, i1));
+            message = message.substring(i1);
+            i = message.length();
+        } while(0 != i);
     }
 
     @Override
