--- a/net/minecraft/util/datafix/fixes/ChunkProtoTickListFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkProtoTickListFix.java
@@ -181,7 +_,7 @@
         int i2 = index >>> 8 & 15;
         String string = idGetter.apply(palette != null ? palette.get().get(i, i1, i2) : null);
         return data.createMap(
-            ImmutableMap.builder()
+            ImmutableMap.<Dynamic<?>, Dynamic<?>>builder()
                 .put(data.createString("i"), data.createString(string))
                 .put(data.createString("x"), data.createInt(x * 16 + i))
                 .put(data.createString("y"), data.createInt(y * 16 + i1))
