--- a/net/minecraft/util/StringRepresentable.java
+++ b/net/minecraft/util/StringRepresentable.java
@@ -29,7 +29,7 @@
         E[] enums = (E[])enumValues.get();
         if (enums.length > 16) {
             Map<String, E> map = Arrays.stream(enums)
-                .collect(Collectors.toMap(_enum -> keyFunction.apply(((StringRepresentable)_enum).getSerializedName()), _enum -> (Enum)_enum));
+                .collect(Collectors.toMap(_enum -> keyFunction.apply(((StringRepresentable)_enum).getSerializedName()), _enum -> _enum));
             return new StringRepresentable.EnumCodec<>(enums, string -> string == null ? null : map.get(string));
         } else {
             return new StringRepresentable.EnumCodec<>(enums, string -> {
